{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "surgical-disease",
   "metadata": {},
   "source": [
    "Reading Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bigger-camel",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import scanpy as sc\n",
    "from glob import iglob\n",
    "import anndata\n",
    "import os\n",
    "import sklearn\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "import pickle\n",
    "import scipy\n",
    "import matplotlib as mpl\n",
    "import matplotlib.pyplot as plt\n",
    "np.random.seed(0)\n",
    "sc.settings.verbosity = 3\n",
    "%matplotlib inline\n",
    "%config InlineBackend.figure_format = 'retina'\n",
    "sc.settings.set_figure_params(dpi = 100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "vanilla-thailand",
   "metadata": {},
   "outputs": [],
   "source": [
    "save_path = '/lustre/scratch117/cellgen/team292/ab55/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "religious-victorian",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 97499 × 22572\n",
       "    obs: 'souporcell', 'demultiplexed', 'sample_names', 'log2p1_count', 'percent_mito', 'n_genes', 'batch', 'FolderName', 'Lane', 'Sort', 'Type', 'Donor Id', 'Age', 'Sex', 'Race', 'Ethnicity', 'BMI', 'Pre-existing heart disease', 'Pre-existing lung disease', 'Pre-existing kidney disease', 'Pre-existing diabetes', 'Pre-existing Hypertension', 'Pre-existing immunocompromised condition', 'Smoking', 'SARS-CoV-2 PCR', 'SARS-CoV-2 Ab', 'Symptomatic', 'Admitted to hospital', 'Highest level of respiratory support', 'Vasoactive agents required during hospitalization', '28-day death', 'scrublet_pred', 'scrublet_local_pred', 'scrublet_score', 'scrublet_cluster_score', 'filtered_cells', 'S_score', 'G2M_score', 'phase', 'leiden_sampl_cc', 'leidenres2_sampl_cc', 'DonorSubset', 'leiden_scvi_subset_cc', 'leidenres2_scvi_subset_cc', 'leiden_tvi_subset_cc', 'leidenres2_tvi_subset_cc', 'louvain', 'Celltype_Predictions', 'Prediction_Probabilities', 'Final_Celltype'\n",
       "    var: 'gene_ids', 'feature_types', 'n_cells'\n",
       "    uns: 'batch_colors', 'louvain', 'louvain_colors', 'neighbors', 'phase_colors', 'sample_names_colors', 't-test', 'umap'\n",
       "    obsm: 'X_umap', 'X_umap_sampl_cc', 'X_umap_scvi_subset_cc', 'X_umap_tvi_subset_cc', 'scvi_sampl_cc', 'scvi_subset_cc', 'tvi_subset_cc'\n",
       "    obsp: 'connectivities', 'distances'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load data\n",
    "adata = anndata.read_h5ad(save_path+\"N3-pbmc-final-clustering.h5ad\")\n",
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "romantic-emerald",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 691683 × 31279\n",
       "    obs: 'sample_id', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden', 'Resample', 'Collection_Day', 'Sex', 'Age', 'Swab_result', 'Status', 'Smoker', 'Status_on_day_collection', 'Status_on_day_collection_summary', 'Days_from_onset', 'Site', 'time_after_LPS', 'Worst_Clinical_Status', 'Outcome', 'initial_clustering', 'full_clustering'\n",
       "    var: 'feature_types'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load reference dataset (Stephenson, E. et al, Nature Medicine, 2021)\n",
    "adata_ref = anndata.read_h5ad(save_path + \"haniffa21_processed_raw.h5ad\")\n",
    "adata_ref"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "under-latin",
   "metadata": {},
   "source": [
    "Processing Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "complete-sight",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "filtered out 11187 genes that are detected in less than 3 cells\n",
      "/opt/conda/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2487: FutureWarning: The `inplace` parameter in pandas.Categorical.remove_unused_categories is deprecated and will be removed in a future version.\n",
      "  res = method(*args, **kwargs)\n",
      "normalizing by total count per cell\n",
      "    finished (0:00:01): normalized adata.X and added    'n_counts', counts per cell before normalization (adata.obs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 97499 × 22572\n",
       "    obs: 'souporcell', 'demultiplexed', 'sample_names', 'log2p1_count', 'percent_mito', 'n_genes', 'batch', 'FolderName', 'Lane', 'Sort', 'Type', 'Donor Id', 'Age', 'Sex', 'Race', 'Ethnicity', 'BMI', 'Pre-existing heart disease', 'Pre-existing lung disease', 'Pre-existing kidney disease', 'Pre-existing diabetes', 'Pre-existing Hypertension', 'Pre-existing immunocompromised condition', 'Smoking', 'SARS-CoV-2 PCR', 'SARS-CoV-2 Ab', 'Symptomatic', 'Admitted to hospital', 'Highest level of respiratory support', 'Vasoactive agents required during hospitalization', '28-day death', 'scrublet_pred', 'scrublet_local_pred', 'scrublet_score', 'scrublet_cluster_score', 'filtered_cells', 'S_score', 'G2M_score', 'phase', 'leiden_sampl_cc', 'leidenres2_sampl_cc', 'DonorSubset', 'leiden_scvi_subset_cc', 'leidenres2_scvi_subset_cc', 'leiden_tvi_subset_cc', 'leidenres2_tvi_subset_cc', 'louvain', 'Celltype_Predictions', 'Prediction_Probabilities', 'Final_Celltype', 'n_counts'\n",
       "    var: 'gene_ids', 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_umap', 'X_umap_sampl_cc', 'X_umap_scvi_subset_cc', 'X_umap_tvi_subset_cc', 'scvi_sampl_cc', 'scvi_subset_cc', 'tvi_subset_cc'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata = anndata.AnnData(X = adata.raw.X, obs = adata.obs, var = adata.raw.var, obsm = adata.obsm).copy()\n",
    "sc.pp.filter_genes(adata, min_cells = 3)\n",
    "sc.pp.normalize_per_cell(adata, counts_per_cell_after = 1e4)\n",
    "sc.pp.log1p(adata)\n",
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "biological-pepper",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2487: FutureWarning: The `inplace` parameter in pandas.Categorical.remove_unused_categories is deprecated and will be removed in a future version.\n",
      "  res = method(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "View of AnnData object with n_obs × n_vars = 97499 × 22553\n",
       "    obs: 'souporcell', 'demultiplexed', 'sample_names', 'log2p1_count', 'percent_mito', 'n_genes', 'batch', 'FolderName', 'Lane', 'Sort', 'Type', 'Donor Id', 'Age', 'Sex', 'Race', 'Ethnicity', 'BMI', 'Pre-existing heart disease', 'Pre-existing lung disease', 'Pre-existing kidney disease', 'Pre-existing diabetes', 'Pre-existing Hypertension', 'Pre-existing immunocompromised condition', 'Smoking', 'SARS-CoV-2 PCR', 'SARS-CoV-2 Ab', 'Symptomatic', 'Admitted to hospital', 'Highest level of respiratory support', 'Vasoactive agents required during hospitalization', '28-day death', 'scrublet_pred', 'scrublet_local_pred', 'scrublet_score', 'scrublet_cluster_score', 'filtered_cells', 'S_score', 'G2M_score', 'phase', 'leiden_sampl_cc', 'leidenres2_sampl_cc', 'DonorSubset', 'leiden_scvi_subset_cc', 'leidenres2_scvi_subset_cc', 'leiden_tvi_subset_cc', 'leidenres2_tvi_subset_cc', 'louvain', 'Celltype_Predictions', 'Prediction_Probabilities', 'Final_Celltype', 'n_counts'\n",
       "    var: 'gene_ids', 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_umap', 'X_umap_sampl_cc', 'X_umap_scvi_subset_cc', 'X_umap_tvi_subset_cc', 'scvi_sampl_cc', 'scvi_subset_cc', 'tvi_subset_cc'"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# using Ensembl Biomart list to filter genes on the Y chromosome\n",
    "Y_chr_table = pd.read_csv('/home/jovyan/COVID/NB5_DE Analysis/Y_chr_genes.txt')\n",
    "Y_chr = list(Y_chr_table['Gene name'])\n",
    "all_genes = adata.var.index\n",
    "genes_to_remove = list(set(all_genes) & set(Y_chr))\n",
    "all_genes = [ele for ele in all_genes if ele not in genes_to_remove]\n",
    "adata = adata[: , all_genes]\n",
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "caroline-pittsburgh",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "192"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# list of surface proteins profiled\n",
    "protein_list = np.array(adata.var_names)[adata.var[\"feature_types\"] ==  \"Antibody Capture\"].tolist()\n",
    "len(protein_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "younger-adult",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Trying to set attribute `.obs` of view, copying.\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>souporcell</th>\n",
       "      <th>demultiplexed</th>\n",
       "      <th>sample_names</th>\n",
       "      <th>log2p1_count</th>\n",
       "      <th>percent_mito</th>\n",
       "      <th>n_genes</th>\n",
       "      <th>batch</th>\n",
       "      <th>FolderName</th>\n",
       "      <th>Lane</th>\n",
       "      <th>Sort</th>\n",
       "      <th>...</th>\n",
       "      <th>Immune_condition</th>\n",
       "      <th>celltype_condition</th>\n",
       "      <th>Immune_condition_RA</th>\n",
       "      <th>celltype_condition_RA</th>\n",
       "      <th>Immune_condition_MS</th>\n",
       "      <th>celltype_condition_MS</th>\n",
       "      <th>Immune_condition_PS</th>\n",
       "      <th>celltype_condition_PS</th>\n",
       "      <th>Immune_condition_Control</th>\n",
       "      <th>celltype_condition_Control</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RV8919578_AAACCTGAGAAACCTA-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "      <td>RV8919578</td>\n",
       "      <td>11.406736</td>\n",
       "      <td>0.017318</td>\n",
       "      <td>1159</td>\n",
       "      <td>2</td>\n",
       "      <td>I-O-2_3_C-0</td>\n",
       "      <td>34527_1</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(bright)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(bright)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>NK CD56(bright)_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8919578_AAACCTGAGAGCAATT-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "      <td>RV8919578</td>\n",
       "      <td>12.939213</td>\n",
       "      <td>0.026614</td>\n",
       "      <td>2058</td>\n",
       "      <td>2</td>\n",
       "      <td>I-O-2_3_C-0</td>\n",
       "      <td>34527_1</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>MS</td>\n",
       "      <td>Naïve B_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>Naïve B_Other</td>\n",
       "      <td>MS</td>\n",
       "      <td>Naïve B_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>Naïve B_Other</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>Naïve B_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8919578_AAACCTGAGAGCTGGT-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "      <td>RV8919578</td>\n",
       "      <td>13.342491</td>\n",
       "      <td>0.007992</td>\n",
       "      <td>2597</td>\n",
       "      <td>2</td>\n",
       "      <td>I-O-2_3_C-0</td>\n",
       "      <td>34527_1</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>MS</td>\n",
       "      <td>pDC_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>pDC_Other</td>\n",
       "      <td>MS</td>\n",
       "      <td>pDC_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>pDC_Other</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>pDC_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8919578_AAACCTGAGCGAAGGG-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "      <td>RV8919578</td>\n",
       "      <td>12.463013</td>\n",
       "      <td>0.031887</td>\n",
       "      <td>1902</td>\n",
       "      <td>2</td>\n",
       "      <td>I-O-2_3_C-0</td>\n",
       "      <td>34527_1</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8919578_AAACCTGAGCGATGAC-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "      <td>RV8919578</td>\n",
       "      <td>11.874598</td>\n",
       "      <td>0.034896</td>\n",
       "      <td>882</td>\n",
       "      <td>2</td>\n",
       "      <td>I-O-2_3_C-0</td>\n",
       "      <td>34527_1</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(bright)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>MS</td>\n",
       "      <td>NK CD56(bright)_MS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>NK CD56(bright)_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8959686_TTTGTCAGTATGAATG-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "      <td>RV8959686</td>\n",
       "      <td>11.049168</td>\n",
       "      <td>0.131728</td>\n",
       "      <td>689</td>\n",
       "      <td>24</td>\n",
       "      <td>I-O-71_C-0</td>\n",
       "      <td>38186</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>PS</td>\n",
       "      <td>CD4 naïve T_PS</td>\n",
       "      <td>Other</td>\n",
       "      <td>CD4 naïve T_Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>CD4 naïve T_Other</td>\n",
       "      <td>PS</td>\n",
       "      <td>CD4 naïve T_PS</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>CD4 naïve T_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8959686_TTTGTCAGTCGGCATC-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "      <td>RV8959686</td>\n",
       "      <td>11.604091</td>\n",
       "      <td>0.026028</td>\n",
       "      <td>1058</td>\n",
       "      <td>24</td>\n",
       "      <td>I-O-71_C-0</td>\n",
       "      <td>38186</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>PS</td>\n",
       "      <td>Naïve B_PS</td>\n",
       "      <td>Other</td>\n",
       "      <td>Naïve B_Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>Naïve B_Other</td>\n",
       "      <td>PS</td>\n",
       "      <td>Naïve B_PS</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>Naïve B_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8959686_TTTGTCAGTGACAAAT-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "      <td>RV8959686</td>\n",
       "      <td>11.200898</td>\n",
       "      <td>0.064173</td>\n",
       "      <td>792</td>\n",
       "      <td>24</td>\n",
       "      <td>I-O-71_C-0</td>\n",
       "      <td>38186</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>PS</td>\n",
       "      <td>NK CD56(bright)_PS</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>NK CD56(bright)_Other</td>\n",
       "      <td>PS</td>\n",
       "      <td>NK CD56(bright)_PS</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>NK CD56(bright)_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8959686_TTTGTCAGTTGCCTCT-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "      <td>RV8959686</td>\n",
       "      <td>12.747355</td>\n",
       "      <td>0.008582</td>\n",
       "      <td>1535</td>\n",
       "      <td>24</td>\n",
       "      <td>I-O-71_C-0</td>\n",
       "      <td>38186</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>PS</td>\n",
       "      <td>CD4 naïve T_PS</td>\n",
       "      <td>Other</td>\n",
       "      <td>CD4 naïve T_Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>CD4 naïve T_Other</td>\n",
       "      <td>PS</td>\n",
       "      <td>CD4 naïve T_PS</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>CD4 naïve T_Diseased</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RV8959686_TTTGTCATCAGTCCCT-1</th>\n",
       "      <td>singlet</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "      <td>RV8959686</td>\n",
       "      <td>13.331057</td>\n",
       "      <td>0.025330</td>\n",
       "      <td>1206</td>\n",
       "      <td>24</td>\n",
       "      <td>I-O-71_C-0</td>\n",
       "      <td>38186</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>PS</td>\n",
       "      <td>Platelets_PS</td>\n",
       "      <td>Other</td>\n",
       "      <td>Platelets_Other</td>\n",
       "      <td>Other</td>\n",
       "      <td>Platelets_Other</td>\n",
       "      <td>PS</td>\n",
       "      <td>Platelets_PS</td>\n",
       "      <td>Diseased</td>\n",
       "      <td>Platelets_Diseased</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>97499 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             souporcell demultiplexed sample_names  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1    singlet    CV19-I-O-3    RV8919578   \n",
       "RV8919578_AAACCTGAGAGCAATT-1    singlet    CV19-I-O-3    RV8919578   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1    singlet    CV19-I-O-2    RV8919578   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1    singlet    CV19-I-O-3    RV8919578   \n",
       "RV8919578_AAACCTGAGCGATGAC-1    singlet    CV19-I-O-2    RV8919578   \n",
       "...                                 ...           ...          ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1    singlet   CV19-I-O-71    RV8959686   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1    singlet   CV19-I-O-71    RV8959686   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1    singlet   CV19-I-O-71    RV8959686   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1    singlet   CV19-I-O-71    RV8959686   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1    singlet   CV19-I-O-71    RV8959686   \n",
       "\n",
       "                              log2p1_count  percent_mito  n_genes batch  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1     11.406736      0.017318     1159     2   \n",
       "RV8919578_AAACCTGAGAGCAATT-1     12.939213      0.026614     2058     2   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1     13.342491      0.007992     2597     2   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1     12.463013      0.031887     1902     2   \n",
       "RV8919578_AAACCTGAGCGATGAC-1     11.874598      0.034896      882     2   \n",
       "...                                    ...           ...      ...   ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1     11.049168      0.131728      689    24   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1     11.604091      0.026028     1058    24   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1     11.200898      0.064173      792    24   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1     12.747355      0.008582     1535    24   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1     13.331057      0.025330     1206    24   \n",
       "\n",
       "                               FolderName     Lane  Sort  ...  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1  I-O-2_3_C-0  34527_1  None  ...   \n",
       "RV8919578_AAACCTGAGAGCAATT-1  I-O-2_3_C-0  34527_1  None  ...   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1  I-O-2_3_C-0  34527_1  None  ...   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1  I-O-2_3_C-0  34527_1  None  ...   \n",
       "RV8919578_AAACCTGAGCGATGAC-1  I-O-2_3_C-0  34527_1  None  ...   \n",
       "...                                   ...      ...   ...  ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1   I-O-71_C-0    38186  None  ...   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1   I-O-71_C-0    38186  None  ...   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1   I-O-71_C-0    38186  None  ...   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1   I-O-71_C-0    38186  None  ...   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1   I-O-71_C-0    38186  None  ...   \n",
       "\n",
       "                             Immune_condition  celltype_condition  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1               MS  NK CD56(bright)_MS   \n",
       "RV8919578_AAACCTGAGAGCAATT-1               MS          Naïve B_MS   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1               MS              pDC_MS   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1               MS     NK CD56(dim)_MS   \n",
       "RV8919578_AAACCTGAGCGATGAC-1               MS  NK CD56(bright)_MS   \n",
       "...                                       ...                 ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1               PS      CD4 naïve T_PS   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1               PS          Naïve B_PS   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1               PS  NK CD56(bright)_PS   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1               PS      CD4 naïve T_PS   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1               PS        Platelets_PS   \n",
       "\n",
       "                              Immune_condition_RA  celltype_condition_RA  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1                Other  NK CD56(bright)_Other   \n",
       "RV8919578_AAACCTGAGAGCAATT-1                Other          Naïve B_Other   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1                Other              pDC_Other   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1                Other     NK CD56(dim)_Other   \n",
       "RV8919578_AAACCTGAGCGATGAC-1                Other  NK CD56(bright)_Other   \n",
       "...                                           ...                    ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1                Other      CD4 naïve T_Other   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1                Other          Naïve B_Other   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1                Other  NK CD56(bright)_Other   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1                Other      CD4 naïve T_Other   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1                Other        Platelets_Other   \n",
       "\n",
       "                             Immune_condition_MS  celltype_condition_MS  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1                  MS     NK CD56(bright)_MS   \n",
       "RV8919578_AAACCTGAGAGCAATT-1                  MS             Naïve B_MS   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1                  MS                 pDC_MS   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1                  MS        NK CD56(dim)_MS   \n",
       "RV8919578_AAACCTGAGCGATGAC-1                  MS     NK CD56(bright)_MS   \n",
       "...                                          ...                    ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1               Other      CD4 naïve T_Other   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1               Other          Naïve B_Other   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1               Other  NK CD56(bright)_Other   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1               Other      CD4 naïve T_Other   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1               Other        Platelets_Other   \n",
       "\n",
       "                             Immune_condition_PS  celltype_condition_PS  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1               Other  NK CD56(bright)_Other   \n",
       "RV8919578_AAACCTGAGAGCAATT-1               Other          Naïve B_Other   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1               Other              pDC_Other   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1               Other     NK CD56(dim)_Other   \n",
       "RV8919578_AAACCTGAGCGATGAC-1               Other  NK CD56(bright)_Other   \n",
       "...                                          ...                    ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1                  PS         CD4 naïve T_PS   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1                  PS             Naïve B_PS   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1                  PS     NK CD56(bright)_PS   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1                  PS         CD4 naïve T_PS   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1                  PS           Platelets_PS   \n",
       "\n",
       "                             Immune_condition_Control  \\\n",
       "RV8919578_AAACCTGAGAAACCTA-1                 Diseased   \n",
       "RV8919578_AAACCTGAGAGCAATT-1                 Diseased   \n",
       "RV8919578_AAACCTGAGAGCTGGT-1                 Diseased   \n",
       "RV8919578_AAACCTGAGCGAAGGG-1                 Diseased   \n",
       "RV8919578_AAACCTGAGCGATGAC-1                 Diseased   \n",
       "...                                               ...   \n",
       "RV8959686_TTTGTCAGTATGAATG-1                 Diseased   \n",
       "RV8959686_TTTGTCAGTCGGCATC-1                 Diseased   \n",
       "RV8959686_TTTGTCAGTGACAAAT-1                 Diseased   \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1                 Diseased   \n",
       "RV8959686_TTTGTCATCAGTCCCT-1                 Diseased   \n",
       "\n",
       "                             celltype_condition_Control  \n",
       "RV8919578_AAACCTGAGAAACCTA-1   NK CD56(bright)_Diseased  \n",
       "RV8919578_AAACCTGAGAGCAATT-1           Naïve B_Diseased  \n",
       "RV8919578_AAACCTGAGAGCTGGT-1               pDC_Diseased  \n",
       "RV8919578_AAACCTGAGCGAAGGG-1      NK CD56(dim)_Diseased  \n",
       "RV8919578_AAACCTGAGCGATGAC-1   NK CD56(bright)_Diseased  \n",
       "...                                                 ...  \n",
       "RV8959686_TTTGTCAGTATGAATG-1       CD4 naïve T_Diseased  \n",
       "RV8959686_TTTGTCAGTCGGCATC-1           Naïve B_Diseased  \n",
       "RV8959686_TTTGTCAGTGACAAAT-1   NK CD56(bright)_Diseased  \n",
       "RV8959686_TTTGTCAGTTGCCTCT-1       CD4 naïve T_Diseased  \n",
       "RV8959686_TTTGTCATCAGTCCCT-1         Platelets_Diseased  \n",
       "\n",
       "[97499 rows x 61 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add donor condition information to be used for the different comparisons\n",
    "donor2condition = {\n",
    "     'CV19-I-O-2': 'MS',\n",
    "     'CV19-I-O-3': 'MS',\n",
    "     'CV19-I-O-73': 'MS',\n",
    "     'CV19-I-O-34': 'PS',\n",
    "     'CV19-I-O-59': 'PS',\n",
    "     'CV19-I-O-71': 'PS',\n",
    "     'CV19-I-O-74': 'PS',\n",
    "     'CV19-I-O-4': 'RA',\n",
    "     'CV19-I-O-39': 'RA',\n",
    "     'CV19-I-O-40': 'RA',\n",
    "     'CV19-I-O-45': 'RA',\n",
    "     'CV19-I-O-51': 'RA',\n",
    "     'CV19-I-O-14': 'Control',\n",
    "     'CV19-I-O-16': 'Control',\n",
    "     'CV19-I-O-17': 'Control',\n",
    "     'CV19-I-O-18': 'Control',\n",
    "     'CV19-I-O-19': 'Control',\n",
    "     'CV19-I-O-20': 'Control',\n",
    "     'CV19-I-O-21': 'Control',\n",
    "     'CV19-I-O-22': 'Control',\n",
    "     'CV19-I-O-23': 'Control',\n",
    "     'CV19-I-O-24': 'Control'}\n",
    "adata.obs['Immune_condition'] = adata.obs['demultiplexed'].map(donor2condition).astype('category')\n",
    "adata.obs['celltype_condition'] = [i + '_' + j for i,j in zip(adata.obs['Final_Celltype'], adata.obs['Immune_condition'])]\n",
    "\n",
    "donor2condition = {\n",
    "     'CV19-I-O-2': 'Other',\n",
    "     'CV19-I-O-3': 'Other',\n",
    "     'CV19-I-O-73': 'Other',\n",
    "     'CV19-I-O-34': 'Other',\n",
    "     'CV19-I-O-59': 'Other',\n",
    "     'CV19-I-O-71': 'Other',\n",
    "     'CV19-I-O-74': 'Other',\n",
    "     'CV19-I-O-4': 'RA',\n",
    "     'CV19-I-O-39': 'RA',\n",
    "     'CV19-I-O-40': 'RA',\n",
    "     'CV19-I-O-45': 'RA',\n",
    "     'CV19-I-O-51': 'RA',\n",
    "     'CV19-I-O-14': 'Other',\n",
    "     'CV19-I-O-16': 'Other',\n",
    "     'CV19-I-O-17': 'Other',\n",
    "     'CV19-I-O-18': 'Other',\n",
    "     'CV19-I-O-19': 'Other',\n",
    "     'CV19-I-O-20': 'Other',\n",
    "     'CV19-I-O-21': 'Other',\n",
    "     'CV19-I-O-22': 'Other',\n",
    "     'CV19-I-O-23': 'Other',\n",
    "     'CV19-I-O-24': 'Other'}\n",
    "adata.obs['Immune_condition_RA'] = adata.obs['demultiplexed'].map(donor2condition).astype('category')\n",
    "adata.obs['celltype_condition_RA'] = [i + '_' + j for i,j in zip(adata.obs['Final_Celltype'], adata.obs['Immune_condition_RA'])]\n",
    "\n",
    "donor2condition = {\n",
    "     'CV19-I-O-2': 'MS',\n",
    "     'CV19-I-O-3': 'MS',\n",
    "     'CV19-I-O-73': 'MS',\n",
    "     'CV19-I-O-34': 'Other',\n",
    "     'CV19-I-O-59': 'Other',\n",
    "     'CV19-I-O-71': 'Other',\n",
    "     'CV19-I-O-74': 'Other',\n",
    "     'CV19-I-O-4': 'Other',\n",
    "     'CV19-I-O-39': 'Other',\n",
    "     'CV19-I-O-40': 'Other',\n",
    "     'CV19-I-O-45': 'Other',\n",
    "     'CV19-I-O-51': 'Other',\n",
    "     'CV19-I-O-14': 'Other',\n",
    "     'CV19-I-O-16': 'Other',\n",
    "     'CV19-I-O-17': 'Other',\n",
    "     'CV19-I-O-18': 'Other',\n",
    "     'CV19-I-O-19': 'Other',\n",
    "     'CV19-I-O-20': 'Other',\n",
    "     'CV19-I-O-21': 'Other',\n",
    "     'CV19-I-O-22': 'Other',\n",
    "     'CV19-I-O-23': 'Other',\n",
    "     'CV19-I-O-24': 'Other'}\n",
    "adata.obs['Immune_condition_MS'] = adata.obs['demultiplexed'].map(donor2condition).astype('category')\n",
    "adata.obs['celltype_condition_MS'] = [i + '_' + j for i,j in zip(adata.obs['Final_Celltype'], adata.obs['Immune_condition_MS'])]\n",
    "\n",
    "donor2condition = {\n",
    "     'CV19-I-O-2': 'Other',\n",
    "     'CV19-I-O-3': 'Other',\n",
    "     'CV19-I-O-73': 'Other',\n",
    "     'CV19-I-O-34': 'PS',\n",
    "     'CV19-I-O-59': 'PS',\n",
    "     'CV19-I-O-71': 'PS',\n",
    "     'CV19-I-O-74': 'PS',\n",
    "     'CV19-I-O-4': 'Other',\n",
    "     'CV19-I-O-39': 'Other',\n",
    "     'CV19-I-O-40': 'Other',\n",
    "     'CV19-I-O-45': 'Other',\n",
    "     'CV19-I-O-51': 'Other',\n",
    "     'CV19-I-O-14': 'Other',\n",
    "     'CV19-I-O-16': 'Other',\n",
    "     'CV19-I-O-17': 'Other',\n",
    "     'CV19-I-O-18': 'Other',\n",
    "     'CV19-I-O-19': 'Other',\n",
    "     'CV19-I-O-20': 'Other',\n",
    "     'CV19-I-O-21': 'Other',\n",
    "     'CV19-I-O-22': 'Other',\n",
    "     'CV19-I-O-23': 'Other',\n",
    "     'CV19-I-O-24': 'Other'}\n",
    "adata.obs['Immune_condition_PS'] = adata.obs['demultiplexed'].map(donor2condition).astype('category')\n",
    "adata.obs['celltype_condition_PS'] = [i + '_' + j for i,j in zip(adata.obs['Final_Celltype'], adata.obs['Immune_condition_PS'])]\n",
    "\n",
    "donor2condition = {\n",
    "     'CV19-I-O-2': 'Diseased',\n",
    "     'CV19-I-O-3': 'Diseased',\n",
    "     'CV19-I-O-73': 'Diseased',\n",
    "     'CV19-I-O-34': 'Diseased',\n",
    "     'CV19-I-O-59': 'Diseased',\n",
    "     'CV19-I-O-71': 'Diseased',\n",
    "     'CV19-I-O-74': 'Diseased',\n",
    "     'CV19-I-O-4': 'Diseased',\n",
    "     'CV19-I-O-39': 'Diseased',\n",
    "     'CV19-I-O-40': 'Diseased',\n",
    "     'CV19-I-O-45': 'Diseased',\n",
    "     'CV19-I-O-51': 'Diseased',\n",
    "     'CV19-I-O-14': 'Control',\n",
    "     'CV19-I-O-16': 'Control',\n",
    "     'CV19-I-O-17': 'Control',\n",
    "     'CV19-I-O-18': 'Control',\n",
    "     'CV19-I-O-19': 'Control',\n",
    "     'CV19-I-O-20': 'Control',\n",
    "     'CV19-I-O-21': 'Control',\n",
    "     'CV19-I-O-22': 'Control',\n",
    "     'CV19-I-O-23': 'Control',\n",
    "     'CV19-I-O-24': 'Control'}\n",
    "adata.obs['Immune_condition_Control'] = adata.obs['demultiplexed'].map(donor2condition).astype('category')\n",
    "adata.obs['celltype_condition_Control'] = [i + '_' + j for i,j in zip(adata.obs['Final_Celltype'], adata.obs['Immune_condition_Control'])]\n",
    "\n",
    "adata.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "popular-hydrogen",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array(['CD14 mono', 'CD16 mono', 'CD4 memory T', 'CD4 naïve T',\n",
       "        'CD8 memory T', 'CD8 naïve T', 'Exhausted B', 'HSC', 'Immature B',\n",
       "        'MAIT', 'Memory B', 'NK CD56(bright)', 'NK CD56(dim)', 'NKT',\n",
       "        'Naïve B', 'Neutrophil', 'Plasma B', 'Plasmablast', 'Platelets',\n",
       "        'Prolif NK', 'Prolif T', 'RBC', 'Treg', 'cDC1', 'cDC2', 'cDC3',\n",
       "        'pDC', 'γδT'], dtype=object),\n",
       " array([23648,  1923,  3276, 26887,  6224,  2387,   510,   270,   255,\n",
       "          223,   457,  3638,  6948,  1807,  8679,   299,   415,  1239,\n",
       "         1444,   226,  1697,  3907,   306,    23,   254,   266,    75,\n",
       "          216]))"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(adata.obs[\"Final_Celltype\"], return_counts = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "prescribed-dairy",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2487: FutureWarning: The `inplace` parameter in pandas.Categorical.remove_unused_categories is deprecated and will be removed in a future version.\n",
      "  res = method(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 6948 × 22553\n",
       "    obs: 'souporcell', 'demultiplexed', 'sample_names', 'log2p1_count', 'percent_mito', 'n_genes', 'batch', 'FolderName', 'Lane', 'Sort', 'Type', 'Donor Id', 'Age', 'Sex', 'Race', 'Ethnicity', 'BMI', 'Pre-existing heart disease', 'Pre-existing lung disease', 'Pre-existing kidney disease', 'Pre-existing diabetes', 'Pre-existing Hypertension', 'Pre-existing immunocompromised condition', 'Smoking', 'SARS-CoV-2 PCR', 'SARS-CoV-2 Ab', 'Symptomatic', 'Admitted to hospital', 'Highest level of respiratory support', 'Vasoactive agents required during hospitalization', '28-day death', 'scrublet_pred', 'scrublet_local_pred', 'scrublet_score', 'scrublet_cluster_score', 'filtered_cells', 'S_score', 'G2M_score', 'phase', 'leiden_sampl_cc', 'leidenres2_sampl_cc', 'DonorSubset', 'leiden_scvi_subset_cc', 'leidenres2_scvi_subset_cc', 'leiden_tvi_subset_cc', 'leidenres2_tvi_subset_cc', 'louvain', 'Celltype_Predictions', 'Prediction_Probabilities', 'Final_Celltype', 'n_counts', 'Immune_condition', 'celltype_condition', 'Immune_condition_RA', 'celltype_condition_RA', 'Immune_condition_MS', 'celltype_condition_MS', 'Immune_condition_PS', 'celltype_condition_PS', 'Immune_condition_Control', 'celltype_condition_Control'\n",
       "    var: 'gene_ids', 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_umap', 'X_umap_sampl_cc', 'X_umap_scvi_subset_cc', 'X_umap_tvi_subset_cc', 'scvi_sampl_cc', 'scvi_subset_cc', 'tvi_subset_cc'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# change for each cell type (example here NK CD56(dim))\n",
    "adata_nkbright = adata[adata.obs['Final_Celltype'].isin([\"NK CD56(dim)\"])].copy()\n",
    "adata_nkbright"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "hollywood-destiny",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "filtered out 6748 genes that are detected in less than 3 cells\n",
      "normalizing by total count per cell\n",
      "    finished (0:00:12): normalized adata.X and added    'n_counts', counts per cell before normalization (adata.obs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 691683 × 24339\n",
       "    obs: 'sample_id', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden', 'Resample', 'Collection_Day', 'Sex', 'Age', 'Swab_result', 'Status', 'Smoker', 'Status_on_day_collection', 'Status_on_day_collection_summary', 'Days_from_onset', 'Site', 'time_after_LPS', 'Worst_Clinical_Status', 'Outcome', 'initial_clustering', 'full_clustering', 'n_counts'\n",
       "    var: 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# process reference dataset similarly and filter\n",
    "adata_ref = anndata.AnnData(X = adata_ref.raw.X, obs = adata_ref.obs, var = adata_ref.raw.var, obsm = adata_ref.obsm).copy()\n",
    "sc.pp.filter_genes(adata_ref, min_cells = 3)\n",
    "sc.pp.normalize_per_cell(adata_ref, counts_per_cell_after = 1e4)\n",
    "sc.pp.log1p(adata_ref)\n",
    "adata_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "applicable-processor",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2487: FutureWarning: The `inplace` parameter in pandas.Categorical.remove_unused_categories is deprecated and will be removed in a future version.\n",
      "  res = method(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "View of AnnData object with n_obs × n_vars = 691683 × 24315\n",
       "    obs: 'sample_id', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden', 'Resample', 'Collection_Day', 'Sex', 'Age', 'Swab_result', 'Status', 'Smoker', 'Status_on_day_collection', 'Status_on_day_collection_summary', 'Days_from_onset', 'Site', 'time_after_LPS', 'Worst_Clinical_Status', 'Outcome', 'initial_clustering', 'full_clustering', 'n_counts'\n",
       "    var: 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_genes_ref = adata_ref.var.index\n",
    "genes_to_remove_ref = list(set(all_genes_ref) & set(Y_chr))\n",
    "all_genes_ref = [ele for ele in all_genes_ref if ele not in genes_to_remove_ref]\n",
    "adata_ref = adata_ref[: , all_genes_ref]\n",
    "adata_ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "awful-enlargement",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 365245 × 24315\n",
       "    obs: 'sample_id', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden', 'Resample', 'Collection_Day', 'Sex', 'Age', 'Swab_result', 'Status', 'Smoker', 'Status_on_day_collection', 'Status_on_day_collection_summary', 'Days_from_onset', 'Site', 'time_after_LPS', 'Worst_Clinical_Status', 'Outcome', 'initial_clustering', 'full_clustering', 'n_counts'\n",
       "    var: 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_ref_COVID = adata_ref[adata_ref.obs[\"Status_on_day_collection_summary\"].isin(['Moderate', 'Critical', 'Severe'])].copy()\n",
    "adata_ref_COVID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "pointed-balloon",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['ASDC', 'B_exhausted', 'B_immature', 'B_malignant', 'B_naive',\n",
       "       'B_non-switched_memory', 'B_switched_memory', 'C1_CD16_mono',\n",
       "       'CD14_mono', 'CD16_mono', 'CD4.CM', 'CD4.EM', 'CD4.IL22',\n",
       "       'CD4.Naive', 'CD4.Prolif', 'CD4.Tfh', 'CD4.Th1', 'CD4.Th17',\n",
       "       'CD4.Th2', 'CD8.EM', 'CD8.Naive', 'CD8.Prolif', 'CD8.TE',\n",
       "       'CD83_CD14_mono', 'DC1', 'DC2', 'DC3', 'DC_prolif', 'HSC',\n",
       "       'ILC1_3', 'ILC2', 'MAIT', 'Mono_prolif', 'NKT', 'NK_16hi',\n",
       "       'NK_56hi', 'NK_prolif', 'Plasma_cell_IgA', 'Plasma_cell_IgG',\n",
       "       'Plasma_cell_IgM', 'Plasmablast', 'Platelets', 'RBC', 'Treg',\n",
       "       'gdT', 'pDC'], dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(adata_ref_COVID.obs[\"full_clustering\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "pursuant-laugh",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "View of AnnData object with n_obs × n_vars = 49571 × 24315\n",
       "    obs: 'sample_id', 'n_genes', 'n_genes_by_counts', 'total_counts', 'total_counts_mt', 'pct_counts_mt', 'leiden', 'Resample', 'Collection_Day', 'Sex', 'Age', 'Swab_result', 'Status', 'Smoker', 'Status_on_day_collection', 'Status_on_day_collection_summary', 'Days_from_onset', 'Site', 'time_after_LPS', 'Worst_Clinical_Status', 'Outcome', 'initial_clustering', 'full_clustering', 'n_counts'\n",
       "    var: 'feature_types', 'n_cells'\n",
       "    uns: 'log1p'\n",
       "    obsm: 'X_pca', 'X_pca_harmony', 'X_umap'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# change for each cell type (example here NK CD56(dim)/NK_16hi)\n",
    "adata_ref_nkbright = adata_ref_COVID[adata_ref_COVID.obs['full_clustering'].isin(['NK_16hi'])]\n",
    "adata_ref_nkbright"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "scientific-lambda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array(['Female', 'Male'], dtype=object), array([5329, 2744]))"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_ref_nkbright = adata_ref_nkbright[adata_ref_nkbright.obs['Site'].isin(['Sanger'])].copy()\n",
    "np.unique(adata_ref_nkbright.obs[\"Sex\"], return_counts = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "controlling-melbourne",
   "metadata": {},
   "source": [
    "limma DE Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "molecular-upgrade",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "visible-tyler",
   "metadata": {},
   "source": [
    "RA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "fabulous-request",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RV8919578_AAACCTGAGCGAAGGG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RV8919578_AAACCTGAGCTGTTCA-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RV8919578_AAACCTGCAGCCAATT-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RV8919578_AAACCTGGTTCCACGG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RV8919578_AAACGGGGTCATATGC-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6943</th>\n",
       "      <td>RV8959686_TTTATGCAGGAGTAGA-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6944</th>\n",
       "      <td>RV8959686_TTTATGCCAGCGATCC-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6945</th>\n",
       "      <td>RV8959686_TTTCCTCGTGCGCTTG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6946</th>\n",
       "      <td>RV8959686_TTTCCTCTCATGTGGT-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6947</th>\n",
       "      <td>RV8959686_TTTGCGCGTGGTAACG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6948 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Cell            Celltype       Sample\n",
       "0     RV8919578_AAACCTGAGCGAAGGG-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "1     RV8919578_AAACCTGAGCTGTTCA-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "2     RV8919578_AAACCTGCAGCCAATT-1  NK CD56(dim)_Other   CV19-I-O-2\n",
       "3     RV8919578_AAACCTGGTTCCACGG-1  NK CD56(dim)_Other   CV19-I-O-2\n",
       "4     RV8919578_AAACGGGGTCATATGC-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "...                            ...                 ...          ...\n",
       "6943  RV8959686_TTTATGCAGGAGTAGA-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6944  RV8959686_TTTATGCCAGCGATCC-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6945  RV8959686_TTTCCTCGTGCGCTTG-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6946  RV8959686_TTTCCTCTCATGTGGT-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6947  RV8959686_TTTGCGCGTGGTAACG-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "\n",
       "[6948 rows x 3 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'NK CD56(dim)_Other'\n",
    "case = 'NK CD56(dim)_RA'\n",
    "\n",
    "t = adata_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_nkbright.obs.index, index = adata_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_nkbright.obs['celltype_condition_RA']],\n",
    "                             'Sample':[str(i) for i in adata_nkbright.obs['demultiplexed']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "vocal-knowing",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NK CD56(dim)_Other', 'NK CD56(dim)_RA'], dtype=object)"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(meta_df.Celltype)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "favorite-laugh",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"NK CD56(dim)_Other\" \"NK CD56(dim)_RA\"   \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 6948shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ctrl\" \"case\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "commercial-cooling",
   "metadata": {},
   "source": [
    "PS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "academic-budapest",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RV8919578_AAACCTGAGCGAAGGG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RV8919578_AAACCTGAGCTGTTCA-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RV8919578_AAACCTGCAGCCAATT-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RV8919578_AAACCTGGTTCCACGG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RV8919578_AAACGGGGTCATATGC-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6943</th>\n",
       "      <td>RV8959686_TTTATGCAGGAGTAGA-1</td>\n",
       "      <td>NK CD56(dim)_PS</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6944</th>\n",
       "      <td>RV8959686_TTTATGCCAGCGATCC-1</td>\n",
       "      <td>NK CD56(dim)_PS</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6945</th>\n",
       "      <td>RV8959686_TTTCCTCGTGCGCTTG-1</td>\n",
       "      <td>NK CD56(dim)_PS</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6946</th>\n",
       "      <td>RV8959686_TTTCCTCTCATGTGGT-1</td>\n",
       "      <td>NK CD56(dim)_PS</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6947</th>\n",
       "      <td>RV8959686_TTTGCGCGTGGTAACG-1</td>\n",
       "      <td>NK CD56(dim)_PS</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6948 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Cell            Celltype       Sample\n",
       "0     RV8919578_AAACCTGAGCGAAGGG-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "1     RV8919578_AAACCTGAGCTGTTCA-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "2     RV8919578_AAACCTGCAGCCAATT-1  NK CD56(dim)_Other   CV19-I-O-2\n",
       "3     RV8919578_AAACCTGGTTCCACGG-1  NK CD56(dim)_Other   CV19-I-O-2\n",
       "4     RV8919578_AAACGGGGTCATATGC-1  NK CD56(dim)_Other   CV19-I-O-3\n",
       "...                            ...                 ...          ...\n",
       "6943  RV8959686_TTTATGCAGGAGTAGA-1     NK CD56(dim)_PS  CV19-I-O-71\n",
       "6944  RV8959686_TTTATGCCAGCGATCC-1     NK CD56(dim)_PS  CV19-I-O-71\n",
       "6945  RV8959686_TTTCCTCGTGCGCTTG-1     NK CD56(dim)_PS  CV19-I-O-71\n",
       "6946  RV8959686_TTTCCTCTCATGTGGT-1     NK CD56(dim)_PS  CV19-I-O-71\n",
       "6947  RV8959686_TTTGCGCGTGGTAACG-1     NK CD56(dim)_PS  CV19-I-O-71\n",
       "\n",
       "[6948 rows x 3 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'NK CD56(dim)_Other'\n",
    "case = 'NK CD56(dim)_PS'\n",
    "\n",
    "t = adata_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_nkbright.obs.index, index = adata_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_nkbright.obs['celltype_condition_PS']],\n",
    "                             'Sample':[str(i) for i in adata_nkbright.obs['demultiplexed']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fourth-prerequisite",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"NK CD56(dim)_Other\" \"NK CD56(dim)_PS\"   \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 6948shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ctrl\" \"case\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "useful-blend",
   "metadata": {},
   "source": [
    "MS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "superior-divorce",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RV8919578_AAACCTGAGCGAAGGG-1</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RV8919578_AAACCTGAGCTGTTCA-1</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RV8919578_AAACCTGCAGCCAATT-1</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RV8919578_AAACCTGGTTCCACGG-1</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RV8919578_AAACGGGGTCATATGC-1</td>\n",
       "      <td>NK CD56(dim)_MS</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6943</th>\n",
       "      <td>RV8959686_TTTATGCAGGAGTAGA-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6944</th>\n",
       "      <td>RV8959686_TTTATGCCAGCGATCC-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6945</th>\n",
       "      <td>RV8959686_TTTCCTCGTGCGCTTG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6946</th>\n",
       "      <td>RV8959686_TTTCCTCTCATGTGGT-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6947</th>\n",
       "      <td>RV8959686_TTTGCGCGTGGTAACG-1</td>\n",
       "      <td>NK CD56(dim)_Other</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6948 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Cell            Celltype       Sample\n",
       "0     RV8919578_AAACCTGAGCGAAGGG-1     NK CD56(dim)_MS   CV19-I-O-3\n",
       "1     RV8919578_AAACCTGAGCTGTTCA-1     NK CD56(dim)_MS   CV19-I-O-3\n",
       "2     RV8919578_AAACCTGCAGCCAATT-1     NK CD56(dim)_MS   CV19-I-O-2\n",
       "3     RV8919578_AAACCTGGTTCCACGG-1     NK CD56(dim)_MS   CV19-I-O-2\n",
       "4     RV8919578_AAACGGGGTCATATGC-1     NK CD56(dim)_MS   CV19-I-O-3\n",
       "...                            ...                 ...          ...\n",
       "6943  RV8959686_TTTATGCAGGAGTAGA-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6944  RV8959686_TTTATGCCAGCGATCC-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6945  RV8959686_TTTCCTCGTGCGCTTG-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6946  RV8959686_TTTCCTCTCATGTGGT-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "6947  RV8959686_TTTGCGCGTGGTAACG-1  NK CD56(dim)_Other  CV19-I-O-71\n",
       "\n",
       "[6948 rows x 3 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'NK CD56(dim)_Other'\n",
    "case = 'NK CD56(dim)_MS'\n",
    "\n",
    "t = adata_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_nkbright.obs.index, index = adata_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_nkbright.obs['celltype_condition_MS']],\n",
    "                             'Sample':[str(i) for i in adata_nkbright.obs['demultiplexed']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "smoking-explosion",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"NK CD56(dim)_MS\"    \"NK CD56(dim)_Other\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 6948shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"case\" \"ctrl\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "unauthorized-deficit",
   "metadata": {},
   "source": [
    "Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "comparable-thong",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "      <th>Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RV8919578_AAACCTGAGCGAAGGG-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RV8919578_AAACCTGAGCTGTTCA-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RV8919578_AAACCTGCAGCCAATT-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RV8919578_AAACCTGGTTCCACGG-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RV8919578_AAACGGGGTCATATGC-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6943</th>\n",
       "      <td>RV8959686_TTTATGCAGGAGTAGA-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6944</th>\n",
       "      <td>RV8959686_TTTATGCCAGCGATCC-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6945</th>\n",
       "      <td>RV8959686_TTTCCTCGTGCGCTTG-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6946</th>\n",
       "      <td>RV8959686_TTTCCTCTCATGTGGT-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6947</th>\n",
       "      <td>RV8959686_TTTGCGCGTGGTAACG-1</td>\n",
       "      <td>NK CD56(dim)_Diseased</td>\n",
       "      <td>CV19-I-O-71</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6948 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Cell               Celltype       Sample\n",
       "0     RV8919578_AAACCTGAGCGAAGGG-1  NK CD56(dim)_Diseased   CV19-I-O-3\n",
       "1     RV8919578_AAACCTGAGCTGTTCA-1  NK CD56(dim)_Diseased   CV19-I-O-3\n",
       "2     RV8919578_AAACCTGCAGCCAATT-1  NK CD56(dim)_Diseased   CV19-I-O-2\n",
       "3     RV8919578_AAACCTGGTTCCACGG-1  NK CD56(dim)_Diseased   CV19-I-O-2\n",
       "4     RV8919578_AAACGGGGTCATATGC-1  NK CD56(dim)_Diseased   CV19-I-O-3\n",
       "...                            ...                    ...          ...\n",
       "6943  RV8959686_TTTATGCAGGAGTAGA-1  NK CD56(dim)_Diseased  CV19-I-O-71\n",
       "6944  RV8959686_TTTATGCCAGCGATCC-1  NK CD56(dim)_Diseased  CV19-I-O-71\n",
       "6945  RV8959686_TTTCCTCGTGCGCTTG-1  NK CD56(dim)_Diseased  CV19-I-O-71\n",
       "6946  RV8959686_TTTCCTCTCATGTGGT-1  NK CD56(dim)_Diseased  CV19-I-O-71\n",
       "6947  RV8959686_TTTGCGCGTGGTAACG-1  NK CD56(dim)_Diseased  CV19-I-O-71\n",
       "\n",
       "[6948 rows x 3 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'NK CD56(dim)_Control'\n",
    "case = 'NK CD56(dim)_Diseased'\n",
    "\n",
    "t = adata_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_nkbright.obs.index, index = adata_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_nkbright.obs['celltype_condition_Control']],\n",
    "                             'Sample':[str(i) for i in adata_nkbright.obs['demultiplexed']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "medium-hypothetical",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"NK CD56(dim)_Diseased\" \"NK CD56(dim)_Control\" \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 6948shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"case\" \"ctrl\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "chronic-manchester",
   "metadata": {},
   "source": [
    "Ref M/F"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "mexican-panama",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>S9_AAAGCAAAGACTAGAT-1</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>S9_AAAGCAAAGGTGCTAG-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>S9_AAAGCAATCATTATCC-1</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>S9_AACCATGGTTGAGTTC-1</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>S9_AACTCTTGTGCTGTAT-1</td>\n",
       "      <td>Male</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8068</th>\n",
       "      <td>S24_TTTACTGGTATGAATG-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8069</th>\n",
       "      <td>S24_TTTCCTCCACACGCTG-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8070</th>\n",
       "      <td>S24_TTTGCGCTCGCAAACT-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8071</th>\n",
       "      <td>S24_TTTGGTTAGTGTCCAT-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8072</th>\n",
       "      <td>S24_TTTGTCAAGCCTATGT-1</td>\n",
       "      <td>Female</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8073 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        Cell Celltype\n",
       "0      S9_AAAGCAAAGACTAGAT-1     Male\n",
       "1      S9_AAAGCAAAGGTGCTAG-1   Female\n",
       "2      S9_AAAGCAATCATTATCC-1     Male\n",
       "3      S9_AACCATGGTTGAGTTC-1     Male\n",
       "4      S9_AACTCTTGTGCTGTAT-1     Male\n",
       "...                      ...      ...\n",
       "8068  S24_TTTACTGGTATGAATG-1   Female\n",
       "8069  S24_TTTCCTCCACACGCTG-1   Female\n",
       "8070  S24_TTTGCGCTCGCAAACT-1   Female\n",
       "8071  S24_TTTGGTTAGTGTCCAT-1   Female\n",
       "8072  S24_TTTGTCAAGCCTATGT-1   Female\n",
       "\n",
       "[8073 rows x 2 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'Male'\n",
    "case = 'Female'\n",
    "\n",
    "t = adata_ref_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_ref_nkbright.obs.index, index = adata_ref_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_ref_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_ref_nkbright.obs['Sex']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "female-enlargement",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Male\"   \"Female\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 8073shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ctrl\" \"case\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./NK CD56(dim)_', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "pending-insider",
   "metadata": {},
   "source": [
    "Ref COVID vs healthy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "electric-height",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.8/site-packages/pandas/core/arrays/categorical.py:2487: FutureWarning: The `inplace` parameter in pandas.Categorical.remove_unused_categories is deprecated and will be removed in a future version.\n",
      "  res = method(*args, **kwargs)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(array(['NK CD56(dim)_COVID', 'NK CD56(dim)_Healthy'], dtype=object),\n",
       " array([3625, 3259]))"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset and format reference object for comparison\n",
    "adata_ref_cam = adata_ref[adata_ref.obs['Site'].isin(['Cambridge'])].copy()\n",
    "adata_ref_all = adata_ref_cam[adata_ref_cam.obs['Status_on_day_collection_summary'].isin(['Critical', 'Healthy', 'Moderate'])].copy()\n",
    "donor2condition = {\n",
    "     'Critical': 'COVID',\n",
    "     'Healthy': 'Healthy',\n",
    "     'Moderate': 'COVID'}\n",
    "adata_ref_all.obs['Condition'] = adata_ref_all.obs['Status_on_day_collection_summary'].map(donor2condition).astype('category')\n",
    "adata_ref_all.obs['celltype_condition'] = [i + '_' + j for i,j in zip(adata_ref_all.obs['full_clustering'], adata_ref_all.obs['Condition'])]\n",
    "\n",
    "adata_ref_all_nkbright = adata_ref_all[adata_ref_all.obs['full_clustering'].isin(['NK_16hi'])].copy()\n",
    "donor2condition = {\n",
    "     'NK_16hi_COVID': 'NK CD56(dim)_COVID',\n",
    "     'NK_16hi_Healthy': 'NK CD56(dim)_Healthy'}\n",
    "adata_ref_all_nkbright.obs['celltype_condition_nkbright'] = adata_ref_all_nkbright.obs['celltype_condition'].map(donor2condition).astype('category')\n",
    "np.unique(adata_ref_all_nkbright.obs['celltype_condition_nkbright'], return_counts = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "specified-belle",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array(['ASDC', 'B_exhausted', 'B_immature', 'B_malignant', 'B_naive',\n",
       "        'B_non-switched_memory', 'B_switched_memory', 'C1_CD16_mono',\n",
       "        'CD14_mono', 'CD16_mono', 'CD4.CM', 'CD4.EM', 'CD4.IL22',\n",
       "        'CD4.Naive', 'CD4.Prolif', 'CD4.Tfh', 'CD4.Th1', 'CD4.Th17',\n",
       "        'CD4.Th2', 'CD8.EM', 'CD8.Naive', 'CD8.Prolif', 'CD8.TE',\n",
       "        'CD83_CD14_mono', 'DC1', 'DC2', 'DC3', 'DC_prolif', 'HSC',\n",
       "        'ILC1_3', 'ILC2', 'MAIT', 'Mono_prolif', 'NKT', 'NK_16hi',\n",
       "        'NK_56hi', 'NK_prolif', 'Plasma_cell_IgA', 'Plasma_cell_IgG',\n",
       "        'Plasma_cell_IgM', 'Plasmablast', 'Platelets', 'RBC', 'Treg',\n",
       "        'gdT', 'pDC'], dtype=object),\n",
       " array([    7,   339,   634,  2313,  4949,   328,   832,     3,   256,\n",
       "          340, 12305,   317,  4818,  6400,    39,  3073,   173,     2,\n",
       "            9,  2178,  4654,    73,  3762,  3597,    35,   169,   427,\n",
       "            2,   160,   208,    20,  2051,     7,   405,  6884,  1940,\n",
       "          187,   297,   266,    37,   497,  2876,  1126,  1283,  2184,\n",
       "          302]))"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.unique(adata_ref_all.obs['full_clustering'], return_counts = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "balanced-partner",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cell</th>\n",
       "      <th>Celltype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BGCV01_AAACCTGAGCTTATCG-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BGCV01_AAACGGGAGATCACGG-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BGCV01_AAAGCAACACGGTGTC-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BGCV01_AAAGCAAGTACCTACA-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BGCV01_AAAGCAAGTTGGTAAA-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6879</th>\n",
       "      <td>BGCV15_TTTACTGTCTCCAACC-1</td>\n",
       "      <td>NK CD56(dim)_COVID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6880</th>\n",
       "      <td>BGCV15_TTTCCTCCACTGTTAG-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6881</th>\n",
       "      <td>BGCV15_TTTGCGCTCACCGTAA-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6882</th>\n",
       "      <td>BGCV15_TTTGGTTTCAAGATCC-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6883</th>\n",
       "      <td>BGCV15_TTTGTCACAAGCCATT-1</td>\n",
       "      <td>NK CD56(dim)_Healthy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6884 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           Cell              Celltype\n",
       "0     BGCV01_AAACCTGAGCTTATCG-1  NK CD56(dim)_Healthy\n",
       "1     BGCV01_AAACGGGAGATCACGG-1  NK CD56(dim)_Healthy\n",
       "2     BGCV01_AAAGCAACACGGTGTC-1  NK CD56(dim)_Healthy\n",
       "3     BGCV01_AAAGCAAGTACCTACA-1  NK CD56(dim)_Healthy\n",
       "4     BGCV01_AAAGCAAGTTGGTAAA-1  NK CD56(dim)_Healthy\n",
       "...                         ...                   ...\n",
       "6879  BGCV15_TTTACTGTCTCCAACC-1    NK CD56(dim)_COVID\n",
       "6880  BGCV15_TTTCCTCCACTGTTAG-1  NK CD56(dim)_Healthy\n",
       "6881  BGCV15_TTTGCGCTCACCGTAA-1  NK CD56(dim)_Healthy\n",
       "6882  BGCV15_TTTGGTTTCAAGATCC-1  NK CD56(dim)_Healthy\n",
       "6883  BGCV15_TTTGTCACAAGCCATT-1  NK CD56(dim)_Healthy\n",
       "\n",
       "[6884 rows x 2 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create dataframes to be used for analysis\n",
    "ctrl = 'NK CD56(dim)_Healthy'\n",
    "case = 'NK CD56(dim)_COVID'\n",
    "\n",
    "t = adata_ref_all_nkbright.X.toarray().T\n",
    "df = pd.DataFrame(data = t, columns = adata_ref_all_nkbright.obs.index, index = adata_ref_all_nkbright.var_names)\n",
    "\n",
    "meta_df = pd.DataFrame(data={'Cell':list(adata_ref_all_nkbright.obs.index),\n",
    "                             'Celltype':[str(i) for i in adata_ref_all_nkbright.obs['celltype_condition_nkbright']]})\n",
    "meta_df.set_index('Cell', inplace = True)\n",
    "meta_df.reset_index(inplace = True)\n",
    "meta_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "baking-literature",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"NK CD56(dim)_Healthy\" \"NK CD56(dim)_COVID\"  \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "R[write to console]: 6884shared cells\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"ctrl\" \"case\"\n"
     ]
    }
   ],
   "source": [
    "%%R -i df -i meta_df -i ctrl -i case\n",
    "\n",
    "#Use edgeR to import, filter, and normalise data then limma (voom) for linear modelling and empirical Bayes moderation to assess DE\n",
    "library(limma)\n",
    "library(edgeR)\n",
    "\n",
    "#Format\n",
    "ex_mat = as.matrix(df)\n",
    "rownames(meta_df) = meta_df$Cell\n",
    "\n",
    "print(unique(meta_df$Celltype))\n",
    "\n",
    "#Shared cells\n",
    "shared_cells = intersect(rownames(meta_df), colnames(ex_mat))\n",
    "message(length(shared_cells), 'shared cells')\n",
    "ex_mat = ex_mat[, shared_cells]\n",
    "meta_df = meta_df[shared_cells,]\n",
    "\n",
    "#Filter lowly expressed genes\n",
    "keep = rowSums(ex_mat, na.rm = T) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "keep = aveLogCPM(ex_mat) > 0.1\n",
    "ex_mat = ex_mat[keep, ]\n",
    "\n",
    "#Extract celltypes\n",
    "cells = rownames(meta_df)\n",
    "celltypes = unique(meta_df$Celltype)\n",
    "covariates = meta_df$covariate\n",
    "\n",
    "#Extract cells according to condition\n",
    "cells_ctrl = rownames(subset(meta_df, Celltype == ctrl))\n",
    "cells_case = rownames(subset(meta_df, Celltype == case))\n",
    "\n",
    "#Build cluster_type vector\n",
    "cluster_type = rep(0, length(cells))\n",
    "names(cluster_type) = cells\n",
    "cluster_type[cells_ctrl] = 'ctrl'\n",
    "cluster_type[cells_case] = 'case'\n",
    "\n",
    "print(unique(cluster_type))\n",
    "design.matrix <- model.matrix(~ 0 + cluster_type)\n",
    "\n",
    "#Define the comparison (i.e., case vs control)\n",
    "contrast.matrix <- makeContrasts(caseVScontrol = cluster_typecase - cluster_typectrl, levels = design.matrix)\n",
    "\n",
    "#Make model and run contrasts\n",
    "fit <- lmFit(ex_mat, design.matrix)\n",
    "fit <- contrasts.fit(fit, contrast.matrix)\n",
    "fit <- eBayes(fit)\n",
    "\n",
    "#Make a dataframe containing the data\n",
    "results = topTable(fit, adjust=\"fdr\", number = nrow(ex_mat), coef = 'caseVScontrol')\n",
    "\n",
    "#Add and filter desired data\n",
    "results$Gene = rownames(results)\n",
    "results = results[ , c('Gene', 'logFC', 'P.Value', 'adj.P.Val')]\n",
    "results$AveExpr_case = apply(ex_mat[results$Gene, cells_case], 1, mean)\n",
    "results$AveExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, mean)\n",
    "results$percentExpr_case = apply(ex_mat[results$Gene, cells_case], 1, function(x) sum(c(x > 0)+0) ) / length(cells_case)\n",
    "results$percentExpr_ctrl = apply(ex_mat[results$Gene, cells_ctrl], 1, function(x) sum(c(x > 0)+0) ) / length(cells_ctrl)\n",
    "results$AveExpr_case = round(results$AveExpr_case, 6)\n",
    "results$AveExpr_ctrl = round(results$AveExpr_ctrl, 6)\n",
    "results$percentExpr_case = round(results$percentExpr_case, 6)\n",
    "results$percentExpr_ctrl = round(results$percentExpr_ctrl, 6)\n",
    "\n",
    "#Store results as a .csv file\n",
    "write.csv(results, file = paste0('./', ctrl, '_vs_', case, '.csv'), row.names = F, col.names = T, quote = F)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "challenging-seven",
   "metadata": {},
   "source": [
    "DE Lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "military-visibility",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_CD11b</th>\n",
       "      <td>-6.050228e-01</td>\n",
       "      <td>6.386196e-291</td>\n",
       "      <td>1.183809e-286</td>\n",
       "      <td>3.845620</td>\n",
       "      <td>4.450643</td>\n",
       "      <td>0.999435</td>\n",
       "      <td>0.999807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TXNIP</th>\n",
       "      <td>-8.537745e-01</td>\n",
       "      <td>2.149126e-203</td>\n",
       "      <td>1.991917e-199</td>\n",
       "      <td>1.829395</td>\n",
       "      <td>2.683170</td>\n",
       "      <td>0.854720</td>\n",
       "      <td>0.949797</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgGFc</th>\n",
       "      <td>-5.947685e-01</td>\n",
       "      <td>4.849857e-196</td>\n",
       "      <td>2.996727e-192</td>\n",
       "      <td>3.247417</td>\n",
       "      <td>3.842186</td>\n",
       "      <td>0.998869</td>\n",
       "      <td>0.999421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD2</th>\n",
       "      <td>9.918110e-01</td>\n",
       "      <td>1.544014e-186</td>\n",
       "      <td>7.155345e-183</td>\n",
       "      <td>3.624814</td>\n",
       "      <td>2.633003</td>\n",
       "      <td>0.973997</td>\n",
       "      <td>0.924310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD45</th>\n",
       "      <td>5.517327e-01</td>\n",
       "      <td>1.573261e-164</td>\n",
       "      <td>5.832708e-161</td>\n",
       "      <td>3.495339</td>\n",
       "      <td>2.943607</td>\n",
       "      <td>0.993782</td>\n",
       "      <td>0.989573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CEP68</th>\n",
       "      <td>-1.928330e-06</td>\n",
       "      <td>9.996357e-01</td>\n",
       "      <td>9.998260e-01</td>\n",
       "      <td>0.020968</td>\n",
       "      <td>0.020970</td>\n",
       "      <td>0.020350</td>\n",
       "      <td>0.018923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL44</th>\n",
       "      <td>-4.217503e-06</td>\n",
       "      <td>9.996768e-01</td>\n",
       "      <td>9.998260e-01</td>\n",
       "      <td>0.144801</td>\n",
       "      <td>0.144806</td>\n",
       "      <td>0.137366</td>\n",
       "      <td>0.133037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERAP1</th>\n",
       "      <td>-3.742003e-06</td>\n",
       "      <td>9.997181e-01</td>\n",
       "      <td>9.998260e-01</td>\n",
       "      <td>0.146676</td>\n",
       "      <td>0.146680</td>\n",
       "      <td>0.139062</td>\n",
       "      <td>0.131493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGHD</th>\n",
       "      <td>-1.890735e-07</td>\n",
       "      <td>9.998774e-01</td>\n",
       "      <td>9.999035e-01</td>\n",
       "      <td>0.001742</td>\n",
       "      <td>0.001742</td>\n",
       "      <td>0.001696</td>\n",
       "      <td>0.001545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TBL1XR1</th>\n",
       "      <td>9.478452e-07</td>\n",
       "      <td>9.999035e-01</td>\n",
       "      <td>9.999035e-01</td>\n",
       "      <td>0.074225</td>\n",
       "      <td>0.074224</td>\n",
       "      <td>0.070096</td>\n",
       "      <td>0.066036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18537 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                 \n",
       "AB_CD11b -6.050228e-01  6.386196e-291  1.183809e-286      3.845620   \n",
       "TXNIP    -8.537745e-01  2.149126e-203  1.991917e-199      1.829395   \n",
       "AB_IgGFc -5.947685e-01  4.849857e-196  2.996727e-192      3.247417   \n",
       "AB_CD2    9.918110e-01  1.544014e-186  7.155345e-183      3.624814   \n",
       "AB_CD45   5.517327e-01  1.573261e-164  5.832708e-161      3.495339   \n",
       "...                ...            ...            ...           ...   \n",
       "CEP68    -1.928330e-06   9.996357e-01   9.998260e-01      0.020968   \n",
       "MRPL44   -4.217503e-06   9.996768e-01   9.998260e-01      0.144801   \n",
       "ERAP1    -3.742003e-06   9.997181e-01   9.998260e-01      0.146676   \n",
       "IGHD     -1.890735e-07   9.998774e-01   9.999035e-01      0.001742   \n",
       "TBL1XR1   9.478452e-07   9.999035e-01   9.999035e-01      0.074225   \n",
       "\n",
       "          AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                        \n",
       "AB_CD11b      4.450643          0.999435          0.999807  \n",
       "TXNIP         2.683170          0.854720          0.949797  \n",
       "AB_IgGFc      3.842186          0.998869          0.999421  \n",
       "AB_CD2        2.633003          0.973997          0.924310  \n",
       "AB_CD45       2.943607          0.993782          0.989573  \n",
       "...                ...               ...               ...  \n",
       "CEP68         0.020970          0.020350          0.018923  \n",
       "MRPL44        0.144806          0.137366          0.133037  \n",
       "ERAP1         0.146680          0.139062          0.131493  \n",
       "IGHD          0.001742          0.001696          0.001545  \n",
       "TBL1XR1       0.074224          0.070096          0.066036  \n",
       "\n",
       "[18537 rows x 7 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read tables and divide into up- and down-regulated genes for filtering\n",
    "DE_table_RA = pd.read_csv('./NK CD56(dim)_Other_vs_NK CD56(dim)_RA.csv', index_col = 0)\n",
    "DE_table_RA_sig_up = DE_table_RA[DE_table_RA['logFC'] > 0]\n",
    "DE_table_RA_sig_down = DE_table_RA[DE_table_RA['logFC'] < 0]\n",
    "DE_table_RA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "intermediate-treat",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_CD99</th>\n",
       "      <td>-8.996892e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.146553</td>\n",
       "      <td>4.046242</td>\n",
       "      <td>0.979845</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD47</th>\n",
       "      <td>-6.969145e-01</td>\n",
       "      <td>1.226373e-284</td>\n",
       "      <td>1.136664e-280</td>\n",
       "      <td>2.799824</td>\n",
       "      <td>3.496739</td>\n",
       "      <td>0.989922</td>\n",
       "      <td>0.999293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCRa/B</th>\n",
       "      <td>-7.652818e-01</td>\n",
       "      <td>3.118925e-272</td>\n",
       "      <td>1.927184e-268</td>\n",
       "      <td>2.418840</td>\n",
       "      <td>3.184122</td>\n",
       "      <td>0.986047</td>\n",
       "      <td>0.997879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD45RA</th>\n",
       "      <td>-8.229667e-01</td>\n",
       "      <td>6.635169e-260</td>\n",
       "      <td>3.074903e-256</td>\n",
       "      <td>4.713449</td>\n",
       "      <td>5.536415</td>\n",
       "      <td>0.999225</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCR</th>\n",
       "      <td>5.567678e-01</td>\n",
       "      <td>9.628503e-239</td>\n",
       "      <td>3.569671e-235</td>\n",
       "      <td>4.605209</td>\n",
       "      <td>4.048441</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STAM-AS1</th>\n",
       "      <td>-2.495571e-06</td>\n",
       "      <td>9.990566e-01</td>\n",
       "      <td>9.992722e-01</td>\n",
       "      <td>0.004342</td>\n",
       "      <td>0.004344</td>\n",
       "      <td>0.003876</td>\n",
       "      <td>0.004242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCK1</th>\n",
       "      <td>9.896183e-06</td>\n",
       "      <td>9.994199e-01</td>\n",
       "      <td>9.995817e-01</td>\n",
       "      <td>0.197461</td>\n",
       "      <td>0.197451</td>\n",
       "      <td>0.171318</td>\n",
       "      <td>0.177625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>E2F8</th>\n",
       "      <td>1.387531e-06</td>\n",
       "      <td>9.995034e-01</td>\n",
       "      <td>9.996113e-01</td>\n",
       "      <td>0.003699</td>\n",
       "      <td>0.003697</td>\n",
       "      <td>0.003101</td>\n",
       "      <td>0.002828</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ST3GAL6</th>\n",
       "      <td>4.240898e-07</td>\n",
       "      <td>9.997001e-01</td>\n",
       "      <td>9.997540e-01</td>\n",
       "      <td>0.001045</td>\n",
       "      <td>0.001044</td>\n",
       "      <td>0.000775</td>\n",
       "      <td>0.000884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOD2-1</th>\n",
       "      <td>1.124900e-07</td>\n",
       "      <td>9.998940e-01</td>\n",
       "      <td>9.998940e-01</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>0.000725</td>\n",
       "      <td>0.000775</td>\n",
       "      <td>0.000707</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18537 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                  \n",
       "AB_CD99   -8.996892e-01   0.000000e+00   0.000000e+00      3.146553   \n",
       "AB_CD47   -6.969145e-01  1.226373e-284  1.136664e-280      2.799824   \n",
       "AB_TCRa/B -7.652818e-01  3.118925e-272  1.927184e-268      2.418840   \n",
       "AB_CD45RA -8.229667e-01  6.635169e-260  3.074903e-256      4.713449   \n",
       "AB_TCR     5.567678e-01  9.628503e-239  3.569671e-235      4.605209   \n",
       "...                 ...            ...            ...           ...   \n",
       "STAM-AS1  -2.495571e-06   9.990566e-01   9.992722e-01      0.004342   \n",
       "NCK1       9.896183e-06   9.994199e-01   9.995817e-01      0.197461   \n",
       "E2F8       1.387531e-06   9.995034e-01   9.996113e-01      0.003699   \n",
       "ST3GAL6    4.240898e-07   9.997001e-01   9.997540e-01      0.001045   \n",
       "SOD2-1     1.124900e-07   9.998940e-01   9.998940e-01      0.000725   \n",
       "\n",
       "           AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                         \n",
       "AB_CD99        4.046242          0.979845          1.000000  \n",
       "AB_CD47        3.496739          0.989922          0.999293  \n",
       "AB_TCRa/B      3.184122          0.986047          0.997879  \n",
       "AB_CD45RA      5.536415          0.999225          1.000000  \n",
       "AB_TCR         4.048441          1.000000          0.999647  \n",
       "...                 ...               ...               ...  \n",
       "STAM-AS1       0.004344          0.003876          0.004242  \n",
       "NCK1           0.197451          0.171318          0.177625  \n",
       "E2F8           0.003697          0.003101          0.002828  \n",
       "ST3GAL6        0.001044          0.000775          0.000884  \n",
       "SOD2-1         0.000725          0.000775          0.000707  \n",
       "\n",
       "[18537 rows x 7 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_PS = pd.read_csv('./NK CD56(dim)_Other_vs_NK CD56(dim)_PS.csv', index_col = 0)\n",
    "DE_table_PS_sig_up = DE_table_PS[DE_table_PS['logFC'] > 0]\n",
    "DE_table_PS_sig_down = DE_table_PS[DE_table_PS['logFC'] < 0]\n",
    "DE_table_PS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "equipped-cooling",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CX3CR1</th>\n",
       "      <td>7.696631e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.945323</td>\n",
       "      <td>0.175660</td>\n",
       "      <td>0.625063</td>\n",
       "      <td>0.151221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD62P(P-Selectin)</th>\n",
       "      <td>-1.303985e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.086339</td>\n",
       "      <td>3.390324</td>\n",
       "      <td>0.934837</td>\n",
       "      <td>0.994549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XAF1</th>\n",
       "      <td>7.137907e-01</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.962077</td>\n",
       "      <td>0.248286</td>\n",
       "      <td>0.595489</td>\n",
       "      <td>0.211791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FCER1G</th>\n",
       "      <td>1.069771e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.904013</td>\n",
       "      <td>0.834242</td>\n",
       "      <td>0.850125</td>\n",
       "      <td>0.485766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CXCR4</th>\n",
       "      <td>-9.809857e-01</td>\n",
       "      <td>2.150426e-284</td>\n",
       "      <td>7.972488e-281</td>\n",
       "      <td>0.809765</td>\n",
       "      <td>1.790750</td>\n",
       "      <td>0.478195</td>\n",
       "      <td>0.841308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP48</th>\n",
       "      <td>7.380527e-06</td>\n",
       "      <td>9.994866e-01</td>\n",
       "      <td>9.996213e-01</td>\n",
       "      <td>0.188081</td>\n",
       "      <td>0.188074</td>\n",
       "      <td>0.167920</td>\n",
       "      <td>0.168181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ANP32E</th>\n",
       "      <td>-9.015551e-06</td>\n",
       "      <td>9.994996e-01</td>\n",
       "      <td>9.996213e-01</td>\n",
       "      <td>0.319128</td>\n",
       "      <td>0.319137</td>\n",
       "      <td>0.277193</td>\n",
       "      <td>0.272158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC010175.1</th>\n",
       "      <td>-3.315463e-07</td>\n",
       "      <td>9.995669e-01</td>\n",
       "      <td>9.996213e-01</td>\n",
       "      <td>0.000545</td>\n",
       "      <td>0.000546</td>\n",
       "      <td>0.000501</td>\n",
       "      <td>0.000606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBE2W</th>\n",
       "      <td>3.879133e-06</td>\n",
       "      <td>9.995674e-01</td>\n",
       "      <td>9.996213e-01</td>\n",
       "      <td>0.070098</td>\n",
       "      <td>0.070094</td>\n",
       "      <td>0.065664</td>\n",
       "      <td>0.066021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IMMP1L</th>\n",
       "      <td>-1.165676e-06</td>\n",
       "      <td>9.998740e-01</td>\n",
       "      <td>9.998740e-01</td>\n",
       "      <td>0.075578</td>\n",
       "      <td>0.075579</td>\n",
       "      <td>0.069173</td>\n",
       "      <td>0.072885</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18537 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             logFC        P.Value      adj.P.Val  \\\n",
       "Gene                                                               \n",
       "CX3CR1                7.696631e-01   0.000000e+00   0.000000e+00   \n",
       "AB_CD62P(P-Selectin) -1.303985e+00   0.000000e+00   0.000000e+00   \n",
       "XAF1                  7.137907e-01   0.000000e+00   0.000000e+00   \n",
       "FCER1G                1.069771e+00   0.000000e+00   0.000000e+00   \n",
       "CXCR4                -9.809857e-01  2.150426e-284  7.972488e-281   \n",
       "...                            ...            ...            ...   \n",
       "USP48                 7.380527e-06   9.994866e-01   9.996213e-01   \n",
       "ANP32E               -9.015551e-06   9.994996e-01   9.996213e-01   \n",
       "AC010175.1           -3.315463e-07   9.995669e-01   9.996213e-01   \n",
       "UBE2W                 3.879133e-06   9.995674e-01   9.996213e-01   \n",
       "IMMP1L               -1.165676e-06   9.998740e-01   9.998740e-01   \n",
       "\n",
       "                      AveExpr_case  AveExpr_ctrl  percentExpr_case  \\\n",
       "Gene                                                                 \n",
       "CX3CR1                    0.945323      0.175660          0.625063   \n",
       "AB_CD62P(P-Selectin)      2.086339      3.390324          0.934837   \n",
       "XAF1                      0.962077      0.248286          0.595489   \n",
       "FCER1G                    1.904013      0.834242          0.850125   \n",
       "CXCR4                     0.809765      1.790750          0.478195   \n",
       "...                            ...           ...               ...   \n",
       "USP48                     0.188081      0.188074          0.167920   \n",
       "ANP32E                    0.319128      0.319137          0.277193   \n",
       "AC010175.1                0.000545      0.000546          0.000501   \n",
       "UBE2W                     0.070098      0.070094          0.065664   \n",
       "IMMP1L                    0.075578      0.075579          0.069173   \n",
       "\n",
       "                      percentExpr_ctrl  \n",
       "Gene                                    \n",
       "CX3CR1                        0.151221  \n",
       "AB_CD62P(P-Selectin)          0.994549  \n",
       "XAF1                          0.211791  \n",
       "FCER1G                        0.485766  \n",
       "CXCR4                         0.841308  \n",
       "...                                ...  \n",
       "USP48                         0.168181  \n",
       "ANP32E                        0.272158  \n",
       "AC010175.1                    0.000606  \n",
       "UBE2W                         0.066021  \n",
       "IMMP1L                        0.072885  \n",
       "\n",
       "[18537 rows x 7 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_MS = pd.read_csv('./NK CD56(dim)_Other_vs_NK CD56(dim)_MS.csv', index_col = 0)\n",
    "DE_table_MS_sig_up = DE_table_MS[DE_table_MS['logFC'] > 0]\n",
    "DE_table_MS_sig_down = DE_table_MS[DE_table_MS['logFC'] < 0]\n",
    "DE_table_MS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "surprising-sixth",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_IgM</th>\n",
       "      <td>-1.479977e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.261228</td>\n",
       "      <td>3.741205</td>\n",
       "      <td>0.979620</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgA</th>\n",
       "      <td>-1.323790e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.538875</td>\n",
       "      <td>3.862665</td>\n",
       "      <td>0.990898</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD235ab</th>\n",
       "      <td>-1.850082e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.175392</td>\n",
       "      <td>3.025474</td>\n",
       "      <td>0.732291</td>\n",
       "      <td>0.989968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgGFc</th>\n",
       "      <td>-1.063513e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.400844</td>\n",
       "      <td>4.464357</td>\n",
       "      <td>0.999011</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_Iglightchain-1</th>\n",
       "      <td>-1.074803e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.011597</td>\n",
       "      <td>4.086401</td>\n",
       "      <td>0.998219</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C20orf197</th>\n",
       "      <td>1.191570e-06</td>\n",
       "      <td>0.999486</td>\n",
       "      <td>0.999618</td>\n",
       "      <td>0.004103</td>\n",
       "      <td>0.004102</td>\n",
       "      <td>0.003759</td>\n",
       "      <td>0.003696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DGKE</th>\n",
       "      <td>-4.285924e-06</td>\n",
       "      <td>0.999517</td>\n",
       "      <td>0.999618</td>\n",
       "      <td>0.064207</td>\n",
       "      <td>0.064211</td>\n",
       "      <td>0.059755</td>\n",
       "      <td>0.058606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRIM36</th>\n",
       "      <td>-4.612535e-07</td>\n",
       "      <td>0.999564</td>\n",
       "      <td>0.999618</td>\n",
       "      <td>0.001042</td>\n",
       "      <td>0.001042</td>\n",
       "      <td>0.001187</td>\n",
       "      <td>0.001056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAD54B</th>\n",
       "      <td>-1.037717e-06</td>\n",
       "      <td>0.999565</td>\n",
       "      <td>0.999618</td>\n",
       "      <td>0.004477</td>\n",
       "      <td>0.004478</td>\n",
       "      <td>0.004155</td>\n",
       "      <td>0.004224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NR4A1</th>\n",
       "      <td>-3.467713e-06</td>\n",
       "      <td>0.999664</td>\n",
       "      <td>0.999664</td>\n",
       "      <td>0.077063</td>\n",
       "      <td>0.077066</td>\n",
       "      <td>0.064306</td>\n",
       "      <td>0.067582</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18537 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                          logFC   P.Value  adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                 \n",
       "AB_IgM            -1.479977e+00  0.000000   0.000000      2.261228   \n",
       "AB_IgA            -1.323790e+00  0.000000   0.000000      2.538875   \n",
       "AB_CD235ab        -1.850082e+00  0.000000   0.000000      1.175392   \n",
       "AB_IgGFc          -1.063513e+00  0.000000   0.000000      3.400844   \n",
       "AB_Iglightchain-1 -1.074803e+00  0.000000   0.000000      3.011597   \n",
       "...                         ...       ...        ...           ...   \n",
       "C20orf197          1.191570e-06  0.999486   0.999618      0.004103   \n",
       "DGKE              -4.285924e-06  0.999517   0.999618      0.064207   \n",
       "TRIM36            -4.612535e-07  0.999564   0.999618      0.001042   \n",
       "RAD54B            -1.037717e-06  0.999565   0.999618      0.004477   \n",
       "NR4A1             -3.467713e-06  0.999664   0.999664      0.077063   \n",
       "\n",
       "                   AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                                 \n",
       "AB_IgM                 3.741205          0.979620          1.000000  \n",
       "AB_IgA                 3.862665          0.990898          1.000000  \n",
       "AB_CD235ab             3.025474          0.732291          0.989968  \n",
       "AB_IgGFc               4.464357          0.999011          1.000000  \n",
       "AB_Iglightchain-1      4.086401          0.998219          1.000000  \n",
       "...                         ...               ...               ...  \n",
       "C20orf197              0.004102          0.003759          0.003696  \n",
       "DGKE                   0.064211          0.059755          0.058606  \n",
       "TRIM36                 0.001042          0.001187          0.001056  \n",
       "RAD54B                 0.004478          0.004155          0.004224  \n",
       "NR4A1                  0.077066          0.064306          0.067582  \n",
       "\n",
       "[18537 rows x 7 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# here with respect to controls\n",
    "DE_table_Control = pd.read_csv('./NK CD56(dim)_Control_vs_NK CD56(dim)_Diseased.csv', index_col = 0)\n",
    "DE_table_Control_sig_up = DE_table_Control[DE_table_Control['logFC'] < 0]\n",
    "DE_table_Control_sig_down = DE_table_Control[DE_table_Control['logFC'] > 0]\n",
    "DE_table_Control"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dietary-booking",
   "metadata": {},
   "source": [
    "Sex filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "excited-ceremony",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RPS4X</th>\n",
       "      <td>7.320186e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.615830</td>\n",
       "      <td>2.883811</td>\n",
       "      <td>0.997561</td>\n",
       "      <td>0.976312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AREG</th>\n",
       "      <td>-1.089285e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.150793</td>\n",
       "      <td>1.240078</td>\n",
       "      <td>0.088009</td>\n",
       "      <td>0.555394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-C</th>\n",
       "      <td>-7.633274e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.125446</td>\n",
       "      <td>3.888773</td>\n",
       "      <td>0.991931</td>\n",
       "      <td>0.998178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DRB1</th>\n",
       "      <td>9.788898e-01</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.303582</td>\n",
       "      <td>0.324692</td>\n",
       "      <td>0.656221</td>\n",
       "      <td>0.181487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCL5</th>\n",
       "      <td>1.012443e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.962843</td>\n",
       "      <td>2.950400</td>\n",
       "      <td>0.977669</td>\n",
       "      <td>0.872085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RPF1</th>\n",
       "      <td>9.158964e-06</td>\n",
       "      <td>0.999366</td>\n",
       "      <td>0.999599</td>\n",
       "      <td>0.176753</td>\n",
       "      <td>0.176744</td>\n",
       "      <td>0.121223</td>\n",
       "      <td>0.113703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TBC1D14</th>\n",
       "      <td>-7.082602e-06</td>\n",
       "      <td>0.999473</td>\n",
       "      <td>0.999647</td>\n",
       "      <td>0.150704</td>\n",
       "      <td>0.150711</td>\n",
       "      <td>0.103397</td>\n",
       "      <td>0.098032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SPACA6</th>\n",
       "      <td>1.874207e-06</td>\n",
       "      <td>0.999546</td>\n",
       "      <td>0.999663</td>\n",
       "      <td>0.012823</td>\n",
       "      <td>0.012822</td>\n",
       "      <td>0.008632</td>\n",
       "      <td>0.008382</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C9orf131</th>\n",
       "      <td>2.460372e-07</td>\n",
       "      <td>0.999710</td>\n",
       "      <td>0.999768</td>\n",
       "      <td>0.000555</td>\n",
       "      <td>0.000555</td>\n",
       "      <td>0.000375</td>\n",
       "      <td>0.000364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENKD1</th>\n",
       "      <td>-2.699578e-07</td>\n",
       "      <td>0.999850</td>\n",
       "      <td>0.999850</td>\n",
       "      <td>0.002516</td>\n",
       "      <td>0.002516</td>\n",
       "      <td>0.001689</td>\n",
       "      <td>0.001822</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17180 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 logFC   P.Value  adj.P.Val  AveExpr_case  AveExpr_ctrl  \\\n",
       "Gene                                                                      \n",
       "RPS4X     7.320186e-01  0.000000   0.000000      3.615830      2.883811   \n",
       "AREG     -1.089285e+00  0.000000   0.000000      0.150793      1.240078   \n",
       "HLA-C    -7.633274e-01  0.000000   0.000000      3.125446      3.888773   \n",
       "HLA-DRB1  9.788898e-01  0.000000   0.000000      1.303582      0.324692   \n",
       "CCL5      1.012443e+00  0.000000   0.000000      3.962843      2.950400   \n",
       "...                ...       ...        ...           ...           ...   \n",
       "RPF1      9.158964e-06  0.999366   0.999599      0.176753      0.176744   \n",
       "TBC1D14  -7.082602e-06  0.999473   0.999647      0.150704      0.150711   \n",
       "SPACA6    1.874207e-06  0.999546   0.999663      0.012823      0.012822   \n",
       "C9orf131  2.460372e-07  0.999710   0.999768      0.000555      0.000555   \n",
       "ENKD1    -2.699578e-07  0.999850   0.999850      0.002516      0.002516   \n",
       "\n",
       "          percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                          \n",
       "RPS4X             0.997561          0.976312  \n",
       "AREG              0.088009          0.555394  \n",
       "HLA-C             0.991931          0.998178  \n",
       "HLA-DRB1          0.656221          0.181487  \n",
       "CCL5              0.977669          0.872085  \n",
       "...                    ...               ...  \n",
       "RPF1              0.121223          0.113703  \n",
       "TBC1D14           0.103397          0.098032  \n",
       "SPACA6            0.008632          0.008382  \n",
       "C9orf131          0.000375          0.000364  \n",
       "ENKD1             0.001689          0.001822  \n",
       "\n",
       "[17180 rows x 7 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter genes differentially expressed either in the same direction for the females, RA, and MS comparisons, or for the males and control comparisons\n",
    "DE_table_Ref = pd.read_csv('./NK CD56(dim)_Male_vs_Female.csv', index_col = 0)\n",
    "DE_table_Ref_sig = DE_table_Ref[DE_table_Ref['adj.P.Val'] < 0.05]\n",
    "DE_table_Ref_sig_up = DE_table_Ref_sig[DE_table_Ref_sig['logFC'] > 0]\n",
    "DE_table_Ref_sig_down = DE_table_Ref_sig[DE_table_Ref_sig['logFC'] < 0]\n",
    "DE_table_Ref"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "banned-accordance",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7671"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_RA_up = []\n",
    "curr_subset_DE = DE_table_RA_sig_up.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_RA_up.append(list(curr_subset_DE.index))\n",
    "list_genes_RA_up = [item for sublist in list_genes_RA_up for item in sublist]\n",
    "len(list_genes_RA_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "hourly-demonstration",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "10866"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_RA_down = []\n",
    "curr_subset_DE = DE_table_RA_sig_down.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_RA_down.append(list(curr_subset_DE.index))\n",
    "list_genes_RA_down = [item for sublist in list_genes_RA_down for item in sublist]\n",
    "len(list_genes_RA_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "surgical-think",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11763"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_MS_up = []\n",
    "curr_subset_DE = DE_table_MS_sig_up.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_MS_up.append(list(curr_subset_DE.index))\n",
    "list_genes_MS_up = [item for sublist in list_genes_MS_up for item in sublist]\n",
    "len(list_genes_MS_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "addressed-multimedia",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6774"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_MS_down = []\n",
    "curr_subset_DE = DE_table_MS_sig_down.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_MS_down.append(list(curr_subset_DE.index))\n",
    "list_genes_MS_down = [item for sublist in list_genes_MS_down for item in sublist]\n",
    "len(list_genes_MS_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "fuzzy-presence",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7227"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_PS_up = []\n",
    "curr_subset_DE = DE_table_PS_sig_up.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_PS_up.append(list(curr_subset_DE.index))\n",
    "list_genes_PS_up = [item for sublist in list_genes_PS_up for item in sublist]\n",
    "len(list_genes_PS_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "stretch-drunk",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11310"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_PS_down = []\n",
    "curr_subset_DE = DE_table_PS_sig_down.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_PS_down.append(list(curr_subset_DE.index))\n",
    "list_genes_PS_down = [item for sublist in list_genes_PS_down for item in sublist]\n",
    "len(list_genes_PS_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "demonstrated-sample",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4813"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_Control_up = []\n",
    "curr_subset_DE = DE_table_Control_sig_up.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_Control_up.append(list(curr_subset_DE.index))\n",
    "list_genes_Control_up = [item for sublist in list_genes_Control_up for item in sublist]\n",
    "len(list_genes_Control_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "hired-folder",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "13724"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_Control_down = []\n",
    "curr_subset_DE = DE_table_Control_sig_down.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_Control_down.append(list(curr_subset_DE.index))\n",
    "list_genes_Control_down = [item for sublist in list_genes_Control_down for item in sublist]\n",
    "len(list_genes_Control_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "silver-metallic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1411"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_ref_up = []\n",
    "curr_subset_DE = DE_table_Ref_sig_up.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_ref_up.append(list(curr_subset_DE.index))\n",
    "list_genes_ref_up = [item for sublist in list_genes_ref_up for item in sublist]\n",
    "len(list_genes_ref_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "compliant-variety",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2270"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_ref_down = []\n",
    "curr_subset_DE = DE_table_Ref_sig_down.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_ref_down.append(list(curr_subset_DE.index))\n",
    "list_genes_ref_down = [item for sublist in list_genes_ref_down for item in sublist]\n",
    "len(list_genes_ref_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "delayed-corrections",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "159"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_down_female = list(set(list_genes_ref_down) & set(list_genes_RA_down) & set(list_genes_MS_down))\n",
    "len(genes_down_female)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "short-synthetic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "323"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_up_female = list(set(list_genes_ref_up) & set(list_genes_RA_up) & set(list_genes_MS_up))\n",
    "len(genes_up_female)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "greatest-decrease",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1011"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_down_male = list(set(list_genes_ref_up) & set(list_genes_Control_down))\n",
    "len(genes_down_male)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "killing-cooperation",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "413"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genes_up_male = list(set(list_genes_ref_down) & set(list_genes_Control_up))\n",
    "len(genes_up_male)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "wound-format",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18055"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "markers_RA_up = [ele for ele in list_genes_RA_up if ele not in genes_up_female]\n",
    "markers_RA_down = [ele for ele in list_genes_RA_down if ele not in genes_down_female]\n",
    "markers_RA = [markers_RA_up, markers_RA_down]\n",
    "markers_RA = [item for sublist in markers_RA for item in sublist]\n",
    "len(markers_RA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "specific-orleans",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18055"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "markers_MS_up = [ele for ele in list_genes_MS_up if ele not in genes_up_female]\n",
    "markers_MS_down = [ele for ele in list_genes_MS_down if ele not in genes_down_female]\n",
    "markers_MS = [markers_MS_up, markers_MS_down]\n",
    "markers_MS = [item for sublist in markers_MS for item in sublist]\n",
    "len(markers_MS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "guided-bermuda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "17113"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "markers_Control_up = [ele for ele in list_genes_Control_up if ele not in genes_up_male]\n",
    "markers_Control_down = [ele for ele in list_genes_Control_down if ele not in genes_down_male]\n",
    "markers_Control = [markers_Control_up, markers_Control_down]\n",
    "markers_Control = [item for sublist in markers_Control for item in sublist]\n",
    "len(markers_Control)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "innovative-brief",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18537"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "markers_PS = [list_genes_PS_up, list_genes_PS_down]\n",
    "markers_PS = [item for sublist in markers_PS for item in sublist]\n",
    "len(markers_PS)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "effective-withdrawal",
   "metadata": {},
   "source": [
    "COVID filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "favorite-princess",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MTRNR2L8</th>\n",
       "      <td>1.132496</td>\n",
       "      <td>7.595587e-246</td>\n",
       "      <td>1.359610e-241</td>\n",
       "      <td>1.505164</td>\n",
       "      <td>0.372668</td>\n",
       "      <td>0.516138</td>\n",
       "      <td>0.205278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-CYB</th>\n",
       "      <td>-0.631185</td>\n",
       "      <td>3.592371e-228</td>\n",
       "      <td>3.215172e-224</td>\n",
       "      <td>2.926788</td>\n",
       "      <td>3.557973</td>\n",
       "      <td>0.964138</td>\n",
       "      <td>0.997238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EEF1A1</th>\n",
       "      <td>-0.314888</td>\n",
       "      <td>2.585867e-221</td>\n",
       "      <td>1.542900e-217</td>\n",
       "      <td>4.344396</td>\n",
       "      <td>4.659284</td>\n",
       "      <td>0.999724</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-CO3</th>\n",
       "      <td>-0.556180</td>\n",
       "      <td>5.807279e-215</td>\n",
       "      <td>2.598757e-211</td>\n",
       "      <td>3.088515</td>\n",
       "      <td>3.644695</td>\n",
       "      <td>0.982897</td>\n",
       "      <td>0.996625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DPB1</th>\n",
       "      <td>0.591804</td>\n",
       "      <td>4.287698e-197</td>\n",
       "      <td>1.534996e-193</td>\n",
       "      <td>0.971850</td>\n",
       "      <td>0.380047</td>\n",
       "      <td>0.593379</td>\n",
       "      <td>0.248236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORO2A</th>\n",
       "      <td>0.009773</td>\n",
       "      <td>1.108312e-02</td>\n",
       "      <td>4.946095e-02</td>\n",
       "      <td>0.024926</td>\n",
       "      <td>0.015152</td>\n",
       "      <td>0.021241</td>\n",
       "      <td>0.011660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AC008966.1</th>\n",
       "      <td>0.005672</td>\n",
       "      <td>1.118312e-02</td>\n",
       "      <td>4.989477e-02</td>\n",
       "      <td>0.009007</td>\n",
       "      <td>0.003335</td>\n",
       "      <td>0.006897</td>\n",
       "      <td>0.002455</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ABCD2</th>\n",
       "      <td>0.008436</td>\n",
       "      <td>1.120967e-02</td>\n",
       "      <td>4.998302e-02</td>\n",
       "      <td>0.018577</td>\n",
       "      <td>0.010141</td>\n",
       "      <td>0.015172</td>\n",
       "      <td>0.008285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BNIP2</th>\n",
       "      <td>-0.042378</td>\n",
       "      <td>1.120990e-02</td>\n",
       "      <td>4.998302e-02</td>\n",
       "      <td>0.448649</td>\n",
       "      <td>0.491027</td>\n",
       "      <td>0.330759</td>\n",
       "      <td>0.332924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TMOD2</th>\n",
       "      <td>0.007107</td>\n",
       "      <td>1.121127e-02</td>\n",
       "      <td>4.998302e-02</td>\n",
       "      <td>0.013828</td>\n",
       "      <td>0.006721</td>\n",
       "      <td>0.011310</td>\n",
       "      <td>0.005216</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4015 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                               \n",
       "MTRNR2L8    1.132496  7.595587e-246  1.359610e-241      1.505164   \n",
       "MT-CYB     -0.631185  3.592371e-228  3.215172e-224      2.926788   \n",
       "EEF1A1     -0.314888  2.585867e-221  1.542900e-217      4.344396   \n",
       "MT-CO3     -0.556180  5.807279e-215  2.598757e-211      3.088515   \n",
       "HLA-DPB1    0.591804  4.287698e-197  1.534996e-193      0.971850   \n",
       "...              ...            ...            ...           ...   \n",
       "CORO2A      0.009773   1.108312e-02   4.946095e-02      0.024926   \n",
       "AC008966.1  0.005672   1.118312e-02   4.989477e-02      0.009007   \n",
       "ABCD2       0.008436   1.120967e-02   4.998302e-02      0.018577   \n",
       "BNIP2      -0.042378   1.120990e-02   4.998302e-02      0.448649   \n",
       "TMOD2       0.007107   1.121127e-02   4.998302e-02      0.013828   \n",
       "\n",
       "            AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                          \n",
       "MTRNR2L8        0.372668          0.516138          0.205278  \n",
       "MT-CYB          3.557973          0.964138          0.997238  \n",
       "EEF1A1          4.659284          0.999724          1.000000  \n",
       "MT-CO3          3.644695          0.982897          0.996625  \n",
       "HLA-DPB1        0.380047          0.593379          0.248236  \n",
       "...                  ...               ...               ...  \n",
       "CORO2A          0.015152          0.021241          0.011660  \n",
       "AC008966.1      0.003335          0.006897          0.002455  \n",
       "ABCD2           0.010141          0.015172          0.008285  \n",
       "BNIP2           0.491027          0.330759          0.332924  \n",
       "TMOD2           0.006721          0.011310          0.005216  \n",
       "\n",
       "[4015 rows x 7 columns]"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# subset genes to those differentially expressed in the COVID-19 vs. healthy comparison\n",
    "DE_table_COVID = pd.read_csv('./NK CD56(dim)_Healthy_vs_NK CD56(dim)_COVID.csv', index_col = 0)\n",
    "DE_table_COVID_sig = DE_table_COVID[DE_table_COVID['adj.P.Val'] < 0.05]\n",
    "DE_table_COVID_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "twelve-sector",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3339"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_COVID_sig_up = DE_table_COVID_sig[DE_table_COVID_sig['logFC'] > 0]\n",
    "DE_table_COVID_sig_down = DE_table_COVID_sig[DE_table_COVID_sig['logFC'] < 0]\n",
    "len(DE_table_COVID_sig_up)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "hungarian-thompson",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "676"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(DE_table_COVID_sig_down)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "ethical-ethernet",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4015"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list_genes_COVID = []\n",
    "curr_subset_DE = DE_table_COVID_sig.sort_values(by = 'adj.P.Val', ascending = True, axis = 0)\n",
    "list_genes_COVID.append(list(curr_subset_DE.index))\n",
    "list_genes_COVID = [item for sublist in list_genes_COVID for item in sublist]\n",
    "len(list_genes_COVID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "toxic-savage",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3738"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COVID_genes_RA = list(set(list_genes_COVID) & set(markers_RA))\n",
    "len(COVID_genes_RA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "automatic-battle",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4005"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COVID_genes_PS = list(set(list_genes_COVID) & set(markers_PS))\n",
    "len(COVID_genes_PS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "premier-vacation",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3738"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COVID_genes_MS = list(set(list_genes_COVID) & set(markers_MS))\n",
    "len(COVID_genes_MS)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "spoken-metallic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3262"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COVID_genes_Ctrl = list(set(list_genes_COVID) & set(markers_Control))\n",
    "len(COVID_genes_Ctrl)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "controlled-complaint",
   "metadata": {},
   "source": [
    "Export lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "detailed-timing",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CXCR4</th>\n",
       "      <td>7.163205e-01</td>\n",
       "      <td>4.628063e-135</td>\n",
       "      <td>7.799128e-132</td>\n",
       "      <td>2.043018</td>\n",
       "      <td>1.326698</td>\n",
       "      <td>0.884681</td>\n",
       "      <td>0.686619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CREM</th>\n",
       "      <td>3.833234e-01</td>\n",
       "      <td>6.705646e-105</td>\n",
       "      <td>6.215128e-102</td>\n",
       "      <td>0.658848</td>\n",
       "      <td>0.275525</td>\n",
       "      <td>0.434709</td>\n",
       "      <td>0.220506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DRB1</th>\n",
       "      <td>3.756096e-01</td>\n",
       "      <td>4.581733e-95</td>\n",
       "      <td>3.266599e-92</td>\n",
       "      <td>0.672785</td>\n",
       "      <td>0.297176</td>\n",
       "      <td>0.456190</td>\n",
       "      <td>0.229967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FCER1G</th>\n",
       "      <td>-5.749392e-01</td>\n",
       "      <td>1.027311e-87</td>\n",
       "      <td>6.801167e-85</td>\n",
       "      <td>0.712852</td>\n",
       "      <td>1.287792</td>\n",
       "      <td>0.421707</td>\n",
       "      <td>0.648002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OASL</th>\n",
       "      <td>2.345290e-01</td>\n",
       "      <td>1.192154e-68</td>\n",
       "      <td>4.804120e-66</td>\n",
       "      <td>0.391186</td>\n",
       "      <td>0.156656</td>\n",
       "      <td>0.322781</td>\n",
       "      <td>0.128403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>F8</th>\n",
       "      <td>-9.514898e-06</td>\n",
       "      <td>9.969648e-01</td>\n",
       "      <td>9.982899e-01</td>\n",
       "      <td>0.007674</td>\n",
       "      <td>0.007683</td>\n",
       "      <td>0.007914</td>\n",
       "      <td>0.007337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MPP6</th>\n",
       "      <td>1.330114e-05</td>\n",
       "      <td>9.975378e-01</td>\n",
       "      <td>9.986152e-01</td>\n",
       "      <td>0.021962</td>\n",
       "      <td>0.021948</td>\n",
       "      <td>0.021481</td>\n",
       "      <td>0.019695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF638</th>\n",
       "      <td>3.047141e-05</td>\n",
       "      <td>9.981750e-01</td>\n",
       "      <td>9.990373e-01</td>\n",
       "      <td>0.241250</td>\n",
       "      <td>0.241219</td>\n",
       "      <td>0.220464</td>\n",
       "      <td>0.208148</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CD38</th>\n",
       "      <td>-1.846278e-05</td>\n",
       "      <td>9.991175e-01</td>\n",
       "      <td>9.995488e-01</td>\n",
       "      <td>0.364248</td>\n",
       "      <td>0.364266</td>\n",
       "      <td>0.296213</td>\n",
       "      <td>0.285576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TBL1XR1</th>\n",
       "      <td>9.478452e-07</td>\n",
       "      <td>9.999035e-01</td>\n",
       "      <td>9.999035e-01</td>\n",
       "      <td>0.074225</td>\n",
       "      <td>0.074224</td>\n",
       "      <td>0.070096</td>\n",
       "      <td>0.066036</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3738 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                 \n",
       "CXCR4     7.163205e-01  4.628063e-135  7.799128e-132      2.043018   \n",
       "CREM      3.833234e-01  6.705646e-105  6.215128e-102      0.658848   \n",
       "HLA-DRB1  3.756096e-01   4.581733e-95   3.266599e-92      0.672785   \n",
       "FCER1G   -5.749392e-01   1.027311e-87   6.801167e-85      0.712852   \n",
       "OASL      2.345290e-01   1.192154e-68   4.804120e-66      0.391186   \n",
       "...                ...            ...            ...           ...   \n",
       "F8       -9.514898e-06   9.969648e-01   9.982899e-01      0.007674   \n",
       "MPP6      1.330114e-05   9.975378e-01   9.986152e-01      0.021962   \n",
       "ZNF638    3.047141e-05   9.981750e-01   9.990373e-01      0.241250   \n",
       "CD38     -1.846278e-05   9.991175e-01   9.995488e-01      0.364248   \n",
       "TBL1XR1   9.478452e-07   9.999035e-01   9.999035e-01      0.074225   \n",
       "\n",
       "          AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                        \n",
       "CXCR4         1.326698          0.884681          0.686619  \n",
       "CREM          0.275525          0.434709          0.220506  \n",
       "HLA-DRB1      0.297176          0.456190          0.229967  \n",
       "FCER1G        1.287792          0.421707          0.648002  \n",
       "OASL          0.156656          0.322781          0.128403  \n",
       "...                ...               ...               ...  \n",
       "F8            0.007683          0.007914          0.007337  \n",
       "MPP6          0.021948          0.021481          0.019695  \n",
       "ZNF638        0.241219          0.220464          0.208148  \n",
       "CD38          0.364266          0.296213          0.285576  \n",
       "TBL1XR1       0.074224          0.070096          0.066036  \n",
       "\n",
       "[3738 rows x 7 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# save filtered and unfiltered tables for genes and proteins\n",
    "DE_table_RA_vs_rest_filtered1 = DE_table_RA[DE_table_RA.index.isin(COVID_genes_RA)]\n",
    "DE_table_RA_vs_rest_filtered1.to_csv(r'/home/jovyan/COVID/NB7_TF/NK CD56(dim)_DE_table_RA_vs_rest_filtered_no_FDR.csv')\n",
    "DE_table_RA_vs_rest_filtered2 = DE_table_RA_vs_rest_filtered1[DE_table_RA_vs_rest_filtered1['adj.P.Val'] < 0.05]\n",
    "DE_table_RA_vs_rest_filtered2.to_csv(r'/home/jovyan/COVID/NB6_CellPhoneDB/DE_Tables/NK CD56(dim)_DE_table_RA_vs_rest_filtered.csv')\n",
    "DE_table_RA_vs_rest_filtered1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "efficient-jamaica",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MT-CO1</th>\n",
       "      <td>0.530429</td>\n",
       "      <td>9.088048e-101</td>\n",
       "      <td>6.239450e-98</td>\n",
       "      <td>3.927353</td>\n",
       "      <td>3.396923</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.996995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IFITM2</th>\n",
       "      <td>-0.625089</td>\n",
       "      <td>2.599286e-94</td>\n",
       "      <td>1.505718e-91</td>\n",
       "      <td>1.839997</td>\n",
       "      <td>2.465086</td>\n",
       "      <td>0.799225</td>\n",
       "      <td>0.928066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S100A8</th>\n",
       "      <td>0.434779</td>\n",
       "      <td>2.198802e-93</td>\n",
       "      <td>1.235127e-90</td>\n",
       "      <td>0.789919</td>\n",
       "      <td>0.355139</td>\n",
       "      <td>0.522481</td>\n",
       "      <td>0.267762</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IFITM1</th>\n",
       "      <td>-0.606526</td>\n",
       "      <td>2.489758e-92</td>\n",
       "      <td>1.318647e-89</td>\n",
       "      <td>2.482334</td>\n",
       "      <td>3.088861</td>\n",
       "      <td>0.895349</td>\n",
       "      <td>0.953694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRF1</th>\n",
       "      <td>-0.566111</td>\n",
       "      <td>3.953153e-69</td>\n",
       "      <td>1.332356e-66</td>\n",
       "      <td>1.681447</td>\n",
       "      <td>2.247558</td>\n",
       "      <td>0.765116</td>\n",
       "      <td>0.898551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BNIP2</th>\n",
       "      <td>-0.000070</td>\n",
       "      <td>9.969278e-01</td>\n",
       "      <td>9.983589e-01</td>\n",
       "      <td>0.414655</td>\n",
       "      <td>0.414725</td>\n",
       "      <td>0.340310</td>\n",
       "      <td>0.354719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAFAH1B1</th>\n",
       "      <td>0.000071</td>\n",
       "      <td>9.969289e-01</td>\n",
       "      <td>9.983589e-01</td>\n",
       "      <td>0.410842</td>\n",
       "      <td>0.410771</td>\n",
       "      <td>0.334884</td>\n",
       "      <td>0.346589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SNF8</th>\n",
       "      <td>-0.000024</td>\n",
       "      <td>9.973549e-01</td>\n",
       "      <td>9.984860e-01</td>\n",
       "      <td>0.053768</td>\n",
       "      <td>0.053793</td>\n",
       "      <td>0.051163</td>\n",
       "      <td>0.051078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSR</th>\n",
       "      <td>-0.000011</td>\n",
       "      <td>9.989572e-01</td>\n",
       "      <td>9.992267e-01</td>\n",
       "      <td>0.075849</td>\n",
       "      <td>0.075861</td>\n",
       "      <td>0.071318</td>\n",
       "      <td>0.069636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCK1</th>\n",
       "      <td>0.000010</td>\n",
       "      <td>9.994199e-01</td>\n",
       "      <td>9.995817e-01</td>\n",
       "      <td>0.197461</td>\n",
       "      <td>0.197451</td>\n",
       "      <td>0.171318</td>\n",
       "      <td>0.177625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4005 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             logFC        P.Value     adj.P.Val  AveExpr_case  AveExpr_ctrl  \\\n",
       "Gene                                                                          \n",
       "MT-CO1    0.530429  9.088048e-101  6.239450e-98      3.927353      3.396923   \n",
       "IFITM2   -0.625089   2.599286e-94  1.505718e-91      1.839997      2.465086   \n",
       "S100A8    0.434779   2.198802e-93  1.235127e-90      0.789919      0.355139   \n",
       "IFITM1   -0.606526   2.489758e-92  1.318647e-89      2.482334      3.088861   \n",
       "PRF1     -0.566111   3.953153e-69  1.332356e-66      1.681447      2.247558   \n",
       "...            ...            ...           ...           ...           ...   \n",
       "BNIP2    -0.000070   9.969278e-01  9.983589e-01      0.414655      0.414725   \n",
       "PAFAH1B1  0.000071   9.969289e-01  9.983589e-01      0.410842      0.410771   \n",
       "SNF8     -0.000024   9.973549e-01  9.984860e-01      0.053768      0.053793   \n",
       "GSR      -0.000011   9.989572e-01  9.992267e-01      0.075849      0.075861   \n",
       "NCK1      0.000010   9.994199e-01  9.995817e-01      0.197461      0.197451   \n",
       "\n",
       "          percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                          \n",
       "MT-CO1            1.000000          0.996995  \n",
       "IFITM2            0.799225          0.928066  \n",
       "S100A8            0.522481          0.267762  \n",
       "IFITM1            0.895349          0.953694  \n",
       "PRF1              0.765116          0.898551  \n",
       "...                    ...               ...  \n",
       "BNIP2             0.340310          0.354719  \n",
       "PAFAH1B1          0.334884          0.346589  \n",
       "SNF8              0.051163          0.051078  \n",
       "GSR               0.071318          0.069636  \n",
       "NCK1              0.171318          0.177625  \n",
       "\n",
       "[4005 rows x 7 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_PS_vs_rest_filtered1 = DE_table_PS[DE_table_PS.index.isin(COVID_genes_PS)]\n",
    "DE_table_PS_vs_rest_filtered1.to_csv(r'/home/jovyan/COVID/NB7_TF/NK CD56(dim)_DE_table_PS_vs_rest_filtered_no_FDR.csv')\n",
    "DE_table_PS_vs_rest_filtered2 = DE_table_PS_vs_rest_filtered1[DE_table_PS_vs_rest_filtered1['adj.P.Val'] < 0.05].copy()\n",
    "DE_table_PS_vs_rest_filtered2.to_csv(r'/home/jovyan/COVID/NB6_CellPhoneDB/DE_Tables/NK CD56(dim)_DE_table_PS_vs_rest_filtered.csv')\n",
    "DE_table_PS_vs_rest_filtered1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "written-company",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CX3CR1</th>\n",
       "      <td>0.769663</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.945323</td>\n",
       "      <td>0.175660</td>\n",
       "      <td>0.625063</td>\n",
       "      <td>0.151221</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>XAF1</th>\n",
       "      <td>0.713791</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.962077</td>\n",
       "      <td>0.248286</td>\n",
       "      <td>0.595489</td>\n",
       "      <td>0.211791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FCER1G</th>\n",
       "      <td>1.069771</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.904013</td>\n",
       "      <td>0.834242</td>\n",
       "      <td>0.850125</td>\n",
       "      <td>0.485766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CXCR4</th>\n",
       "      <td>-0.980986</td>\n",
       "      <td>2.150426e-284</td>\n",
       "      <td>7.972488e-281</td>\n",
       "      <td>0.809765</td>\n",
       "      <td>1.790750</td>\n",
       "      <td>0.478195</td>\n",
       "      <td>0.841308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CORO1A</th>\n",
       "      <td>0.801174</td>\n",
       "      <td>3.541870e-261</td>\n",
       "      <td>1.094261e-257</td>\n",
       "      <td>2.056371</td>\n",
       "      <td>1.255197</td>\n",
       "      <td>0.929323</td>\n",
       "      <td>0.750252</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TRIM39</th>\n",
       "      <td>0.000014</td>\n",
       "      <td>9.973227e-01</td>\n",
       "      <td>9.985077e-01</td>\n",
       "      <td>0.021667</td>\n",
       "      <td>0.021653</td>\n",
       "      <td>0.020551</td>\n",
       "      <td>0.020594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL50</th>\n",
       "      <td>0.000017</td>\n",
       "      <td>9.980195e-01</td>\n",
       "      <td>9.988817e-01</td>\n",
       "      <td>0.068360</td>\n",
       "      <td>0.068342</td>\n",
       "      <td>0.065163</td>\n",
       "      <td>0.064809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADO</th>\n",
       "      <td>0.000003</td>\n",
       "      <td>9.991192e-01</td>\n",
       "      <td>9.994542e-01</td>\n",
       "      <td>0.011510</td>\n",
       "      <td>0.011507</td>\n",
       "      <td>0.010025</td>\n",
       "      <td>0.010701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VPS51</th>\n",
       "      <td>-0.000015</td>\n",
       "      <td>9.991307e-01</td>\n",
       "      <td>9.994542e-01</td>\n",
       "      <td>0.315330</td>\n",
       "      <td>0.315346</td>\n",
       "      <td>0.272180</td>\n",
       "      <td>0.278417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>USP48</th>\n",
       "      <td>0.000007</td>\n",
       "      <td>9.994866e-01</td>\n",
       "      <td>9.996213e-01</td>\n",
       "      <td>0.188081</td>\n",
       "      <td>0.188074</td>\n",
       "      <td>0.167920</td>\n",
       "      <td>0.168181</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3738 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           logFC        P.Value      adj.P.Val  AveExpr_case  AveExpr_ctrl  \\\n",
       "Gene                                                                         \n",
       "CX3CR1  0.769663   0.000000e+00   0.000000e+00      0.945323      0.175660   \n",
       "XAF1    0.713791   0.000000e+00   0.000000e+00      0.962077      0.248286   \n",
       "FCER1G  1.069771   0.000000e+00   0.000000e+00      1.904013      0.834242   \n",
       "CXCR4  -0.980986  2.150426e-284  7.972488e-281      0.809765      1.790750   \n",
       "CORO1A  0.801174  3.541870e-261  1.094261e-257      2.056371      1.255197   \n",
       "...          ...            ...            ...           ...           ...   \n",
       "TRIM39  0.000014   9.973227e-01   9.985077e-01      0.021667      0.021653   \n",
       "MRPL50  0.000017   9.980195e-01   9.988817e-01      0.068360      0.068342   \n",
       "ADO     0.000003   9.991192e-01   9.994542e-01      0.011510      0.011507   \n",
       "VPS51  -0.000015   9.991307e-01   9.994542e-01      0.315330      0.315346   \n",
       "USP48   0.000007   9.994866e-01   9.996213e-01      0.188081      0.188074   \n",
       "\n",
       "        percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                        \n",
       "CX3CR1          0.625063          0.151221  \n",
       "XAF1            0.595489          0.211791  \n",
       "FCER1G          0.850125          0.485766  \n",
       "CXCR4           0.478195          0.841308  \n",
       "CORO1A          0.929323          0.750252  \n",
       "...                  ...               ...  \n",
       "TRIM39          0.020551          0.020594  \n",
       "MRPL50          0.065163          0.064809  \n",
       "ADO             0.010025          0.010701  \n",
       "VPS51           0.272180          0.278417  \n",
       "USP48           0.167920          0.168181  \n",
       "\n",
       "[3738 rows x 7 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_MS_vs_rest_filtered1 = DE_table_MS[DE_table_MS.index.isin(COVID_genes_MS)]\n",
    "DE_table_MS_vs_rest_filtered1.to_csv(r'/home/jovyan/COVID/NB7_TF/NK CD56(dim)_DE_table_MS_vs_rest_filtered_no_FDR.csv')\n",
    "DE_table_MS_vs_rest_filtered2 = DE_table_MS_vs_rest_filtered1[DE_table_MS_vs_rest_filtered1['adj.P.Val'] < 0.05].copy()\n",
    "DE_table_MS_vs_rest_filtered2.to_csv(r'/home/jovyan/COVID/NB6_CellPhoneDB/DE_Tables/NK CD56(dim)_DE_table_MS_vs_rest_filtered.csv')\n",
    "DE_table_MS_vs_rest_filtered1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "handed-discretion",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>IGLV3-25</th>\n",
       "      <td>-0.329341</td>\n",
       "      <td>1.091939e-243</td>\n",
       "      <td>2.249031e-240</td>\n",
       "      <td>0.022620</td>\n",
       "      <td>0.351961</td>\n",
       "      <td>0.018401</td>\n",
       "      <td>0.269799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGLV1-51</th>\n",
       "      <td>-0.198186</td>\n",
       "      <td>1.988791e-145</td>\n",
       "      <td>2.457749e-142</td>\n",
       "      <td>0.006185</td>\n",
       "      <td>0.204372</td>\n",
       "      <td>0.005342</td>\n",
       "      <td>0.151003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-CO1</th>\n",
       "      <td>0.549465</td>\n",
       "      <td>6.194110e-143</td>\n",
       "      <td>7.176264e-140</td>\n",
       "      <td>3.645188</td>\n",
       "      <td>3.095723</td>\n",
       "      <td>0.999209</td>\n",
       "      <td>0.993136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-ATP6</th>\n",
       "      <td>0.541811</td>\n",
       "      <td>1.266430e-122</td>\n",
       "      <td>1.304212e-119</td>\n",
       "      <td>2.300947</td>\n",
       "      <td>1.759136</td>\n",
       "      <td>0.962604</td>\n",
       "      <td>0.885428</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IGHV5-51</th>\n",
       "      <td>-0.156075</td>\n",
       "      <td>3.941200e-115</td>\n",
       "      <td>3.320819e-112</td>\n",
       "      <td>0.009388</td>\n",
       "      <td>0.165463</td>\n",
       "      <td>0.007915</td>\n",
       "      <td>0.132524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CSRP1</th>\n",
       "      <td>-0.000168</td>\n",
       "      <td>9.858734e-01</td>\n",
       "      <td>9.911186e-01</td>\n",
       "      <td>0.125483</td>\n",
       "      <td>0.125651</td>\n",
       "      <td>0.119114</td>\n",
       "      <td>0.116156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SMIM12</th>\n",
       "      <td>-0.000157</td>\n",
       "      <td>9.879598e-01</td>\n",
       "      <td>9.925109e-01</td>\n",
       "      <td>0.154339</td>\n",
       "      <td>0.154497</td>\n",
       "      <td>0.145231</td>\n",
       "      <td>0.138860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BARD1</th>\n",
       "      <td>0.000031</td>\n",
       "      <td>9.959457e-01</td>\n",
       "      <td>9.978297e-01</td>\n",
       "      <td>0.047249</td>\n",
       "      <td>0.047218</td>\n",
       "      <td>0.043332</td>\n",
       "      <td>0.042767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ATG13</th>\n",
       "      <td>0.000042</td>\n",
       "      <td>9.967040e-01</td>\n",
       "      <td>9.983363e-01</td>\n",
       "      <td>0.144692</td>\n",
       "      <td>0.144650</td>\n",
       "      <td>0.134943</td>\n",
       "      <td>0.131996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CERS5</th>\n",
       "      <td>-0.000022</td>\n",
       "      <td>9.978707e-01</td>\n",
       "      <td>9.988406e-01</td>\n",
       "      <td>0.095857</td>\n",
       "      <td>0.095879</td>\n",
       "      <td>0.092006</td>\n",
       "      <td>0.087645</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3262 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             logFC        P.Value      adj.P.Val  AveExpr_case  AveExpr_ctrl  \\\n",
       "Gene                                                                           \n",
       "IGLV3-25 -0.329341  1.091939e-243  2.249031e-240      0.022620      0.351961   \n",
       "IGLV1-51 -0.198186  1.988791e-145  2.457749e-142      0.006185      0.204372   \n",
       "MT-CO1    0.549465  6.194110e-143  7.176264e-140      3.645188      3.095723   \n",
       "MT-ATP6   0.541811  1.266430e-122  1.304212e-119      2.300947      1.759136   \n",
       "IGHV5-51 -0.156075  3.941200e-115  3.320819e-112      0.009388      0.165463   \n",
       "...            ...            ...            ...           ...           ...   \n",
       "CSRP1    -0.000168   9.858734e-01   9.911186e-01      0.125483      0.125651   \n",
       "SMIM12   -0.000157   9.879598e-01   9.925109e-01      0.154339      0.154497   \n",
       "BARD1     0.000031   9.959457e-01   9.978297e-01      0.047249      0.047218   \n",
       "ATG13     0.000042   9.967040e-01   9.983363e-01      0.144692      0.144650   \n",
       "CERS5    -0.000022   9.978707e-01   9.988406e-01      0.095857      0.095879   \n",
       "\n",
       "          percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                          \n",
       "IGLV3-25          0.018401          0.269799  \n",
       "IGLV1-51          0.005342          0.151003  \n",
       "MT-CO1            0.999209          0.993136  \n",
       "MT-ATP6           0.962604          0.885428  \n",
       "IGHV5-51          0.007915          0.132524  \n",
       "...                    ...               ...  \n",
       "CSRP1             0.119114          0.116156  \n",
       "SMIM12            0.145231          0.138860  \n",
       "BARD1             0.043332          0.042767  \n",
       "ATG13             0.134943          0.131996  \n",
       "CERS5             0.092006          0.087645  \n",
       "\n",
       "[3262 rows x 7 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_Control_vs_rest_filtered1 = DE_table_Control[DE_table_Control.index.isin(COVID_genes_Ctrl)]\n",
    "DE_table_Control_vs_rest_filtered1.to_csv(r'/home/jovyan/COVID/NB7_TF/NK CD56(dim)_DE_table_Control_vs_rest_filtered_no_FDR.csv')\n",
    "DE_table_Control_vs_rest_filtered2 = DE_table_Control_vs_rest_filtered1[DE_table_Control_vs_rest_filtered1['adj.P.Val'] < 0.05].copy()\n",
    "DE_table_Control_vs_rest_filtered2.to_csv(r'/home/jovyan/COVID/NB6_CellPhoneDB/DE_Tables/NK CD56(dim)_DE_table_Control_vs_rest_filtered.csv')\n",
    "DE_table_Control_vs_rest_filtered1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "limiting-invalid",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_CD11b</th>\n",
       "      <td>-0.605023</td>\n",
       "      <td>6.386196e-291</td>\n",
       "      <td>1.183809e-286</td>\n",
       "      <td>3.845620</td>\n",
       "      <td>4.450643</td>\n",
       "      <td>0.999435</td>\n",
       "      <td>0.999807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgGFc</th>\n",
       "      <td>-0.594769</td>\n",
       "      <td>4.849857e-196</td>\n",
       "      <td>2.996727e-192</td>\n",
       "      <td>3.247417</td>\n",
       "      <td>3.842186</td>\n",
       "      <td>0.998869</td>\n",
       "      <td>0.999421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD2</th>\n",
       "      <td>0.991811</td>\n",
       "      <td>1.544014e-186</td>\n",
       "      <td>7.155345e-183</td>\n",
       "      <td>3.624814</td>\n",
       "      <td>2.633003</td>\n",
       "      <td>0.973997</td>\n",
       "      <td>0.924310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD45</th>\n",
       "      <td>0.551733</td>\n",
       "      <td>1.573261e-164</td>\n",
       "      <td>5.832708e-161</td>\n",
       "      <td>3.495339</td>\n",
       "      <td>2.943607</td>\n",
       "      <td>0.993782</td>\n",
       "      <td>0.989573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCRa/B</th>\n",
       "      <td>0.515205</td>\n",
       "      <td>8.362046e-150</td>\n",
       "      <td>2.583454e-146</td>\n",
       "      <td>3.426066</td>\n",
       "      <td>2.910861</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.994207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD49b</th>\n",
       "      <td>0.008006</td>\n",
       "      <td>8.038292e-01</td>\n",
       "      <td>8.905972e-01</td>\n",
       "      <td>1.947126</td>\n",
       "      <td>1.939120</td>\n",
       "      <td>0.868287</td>\n",
       "      <td>0.838965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_Mouse_IgG2b</th>\n",
       "      <td>-0.000640</td>\n",
       "      <td>8.252205e-01</td>\n",
       "      <td>9.023767e-01</td>\n",
       "      <td>0.010074</td>\n",
       "      <td>0.010714</td>\n",
       "      <td>0.010741</td>\n",
       "      <td>0.010234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD158b(KIR2DL2)</th>\n",
       "      <td>0.003851</td>\n",
       "      <td>8.788678e-01</td>\n",
       "      <td>9.340427e-01</td>\n",
       "      <td>2.914367</td>\n",
       "      <td>2.910515</td>\n",
       "      <td>0.987564</td>\n",
       "      <td>0.993242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_GARP(LRRC32)</th>\n",
       "      <td>0.003361</td>\n",
       "      <td>8.818273e-01</td>\n",
       "      <td>9.354182e-01</td>\n",
       "      <td>1.800578</td>\n",
       "      <td>1.797216</td>\n",
       "      <td>0.918033</td>\n",
       "      <td>0.905387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD98</th>\n",
       "      <td>0.000407</td>\n",
       "      <td>9.850340e-01</td>\n",
       "      <td>9.916310e-01</td>\n",
       "      <td>1.252845</td>\n",
       "      <td>1.252438</td>\n",
       "      <td>0.804409</td>\n",
       "      <td>0.775053</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                       \n",
       "AB_CD11b           -0.605023  6.386196e-291  1.183809e-286      3.845620   \n",
       "AB_IgGFc           -0.594769  4.849857e-196  2.996727e-192      3.247417   \n",
       "AB_CD2              0.991811  1.544014e-186  7.155345e-183      3.624814   \n",
       "AB_CD45             0.551733  1.573261e-164  5.832708e-161      3.495339   \n",
       "AB_TCRa/B           0.515205  8.362046e-150  2.583454e-146      3.426066   \n",
       "...                      ...            ...            ...           ...   \n",
       "AB_CD49b            0.008006   8.038292e-01   8.905972e-01      1.947126   \n",
       "AB_Mouse_IgG2b     -0.000640   8.252205e-01   9.023767e-01      0.010074   \n",
       "AB_CD158b(KIR2DL2)  0.003851   8.788678e-01   9.340427e-01      2.914367   \n",
       "AB_GARP(LRRC32)     0.003361   8.818273e-01   9.354182e-01      1.800578   \n",
       "AB_CD98             0.000407   9.850340e-01   9.916310e-01      1.252845   \n",
       "\n",
       "                    AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                                  \n",
       "AB_CD11b                4.450643          0.999435          0.999807  \n",
       "AB_IgGFc                3.842186          0.998869          0.999421  \n",
       "AB_CD2                  2.633003          0.973997          0.924310  \n",
       "AB_CD45                 2.943607          0.993782          0.989573  \n",
       "AB_TCRa/B               2.910861          1.000000          0.994207  \n",
       "...                          ...               ...               ...  \n",
       "AB_CD49b                1.939120          0.868287          0.838965  \n",
       "AB_Mouse_IgG2b          0.010714          0.010741          0.010234  \n",
       "AB_CD158b(KIR2DL2)      2.910515          0.987564          0.993242  \n",
       "AB_GARP(LRRC32)         1.797216          0.918033          0.905387  \n",
       "AB_CD98                 1.252438          0.804409          0.775053  \n",
       "\n",
       "[192 rows x 7 columns]"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_RA_protein = DE_table_RA[DE_table_RA.index.isin(protein_list)]\n",
    "DE_table_RA_protein.to_csv('/home/jovyan/COVID/NB5_DE Analysis/DE proteins/NK CD56(dim)_DE_table_RA_vs_rest_protein.csv')\n",
    "DE_table_RA_protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "israeli-absence",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_CD99</th>\n",
       "      <td>-0.899689</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>3.146553</td>\n",
       "      <td>4.046242</td>\n",
       "      <td>0.979845</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD47</th>\n",
       "      <td>-0.696915</td>\n",
       "      <td>1.226373e-284</td>\n",
       "      <td>1.136664e-280</td>\n",
       "      <td>2.799824</td>\n",
       "      <td>3.496739</td>\n",
       "      <td>0.989922</td>\n",
       "      <td>0.999293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCRa/B</th>\n",
       "      <td>-0.765282</td>\n",
       "      <td>3.118925e-272</td>\n",
       "      <td>1.927184e-268</td>\n",
       "      <td>2.418840</td>\n",
       "      <td>3.184122</td>\n",
       "      <td>0.986047</td>\n",
       "      <td>0.997879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD45RA</th>\n",
       "      <td>-0.822967</td>\n",
       "      <td>6.635169e-260</td>\n",
       "      <td>3.074903e-256</td>\n",
       "      <td>4.713449</td>\n",
       "      <td>5.536415</td>\n",
       "      <td>0.999225</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCR</th>\n",
       "      <td>0.556768</td>\n",
       "      <td>9.628503e-239</td>\n",
       "      <td>3.569671e-235</td>\n",
       "      <td>4.605209</td>\n",
       "      <td>4.048441</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.999647</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD71</th>\n",
       "      <td>-0.004517</td>\n",
       "      <td>8.707759e-01</td>\n",
       "      <td>9.308329e-01</td>\n",
       "      <td>1.401954</td>\n",
       "      <td>1.406471</td>\n",
       "      <td>0.817829</td>\n",
       "      <td>0.810887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_HLA-A2</th>\n",
       "      <td>0.001057</td>\n",
       "      <td>8.896420e-01</td>\n",
       "      <td>9.410976e-01</td>\n",
       "      <td>0.057584</td>\n",
       "      <td>0.056526</td>\n",
       "      <td>0.051163</td>\n",
       "      <td>0.053022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_mouse/ratCD278(ICOS)</th>\n",
       "      <td>0.002633</td>\n",
       "      <td>9.208878e-01</td>\n",
       "      <td>9.582815e-01</td>\n",
       "      <td>2.088329</td>\n",
       "      <td>2.085696</td>\n",
       "      <td>0.959690</td>\n",
       "      <td>0.952633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD194(CCR4)</th>\n",
       "      <td>0.001422</td>\n",
       "      <td>9.541954e-01</td>\n",
       "      <td>9.774490e-01</td>\n",
       "      <td>1.899795</td>\n",
       "      <td>1.898372</td>\n",
       "      <td>0.927907</td>\n",
       "      <td>0.927890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD26</th>\n",
       "      <td>0.000900</td>\n",
       "      <td>9.722646e-01</td>\n",
       "      <td>9.870677e-01</td>\n",
       "      <td>1.514848</td>\n",
       "      <td>1.513948</td>\n",
       "      <td>0.850388</td>\n",
       "      <td>0.856486</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                            \n",
       "AB_CD99                 -0.899689   0.000000e+00   0.000000e+00      3.146553   \n",
       "AB_CD47                 -0.696915  1.226373e-284  1.136664e-280      2.799824   \n",
       "AB_TCRa/B               -0.765282  3.118925e-272  1.927184e-268      2.418840   \n",
       "AB_CD45RA               -0.822967  6.635169e-260  3.074903e-256      4.713449   \n",
       "AB_TCR                   0.556768  9.628503e-239  3.569671e-235      4.605209   \n",
       "...                           ...            ...            ...           ...   \n",
       "AB_CD71                 -0.004517   8.707759e-01   9.308329e-01      1.401954   \n",
       "AB_HLA-A2                0.001057   8.896420e-01   9.410976e-01      0.057584   \n",
       "AB_mouse/ratCD278(ICOS)  0.002633   9.208878e-01   9.582815e-01      2.088329   \n",
       "AB_CD194(CCR4)           0.001422   9.541954e-01   9.774490e-01      1.899795   \n",
       "AB_CD26                  0.000900   9.722646e-01   9.870677e-01      1.514848   \n",
       "\n",
       "                         AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                                       \n",
       "AB_CD99                      4.046242          0.979845          1.000000  \n",
       "AB_CD47                      3.496739          0.989922          0.999293  \n",
       "AB_TCRa/B                    3.184122          0.986047          0.997879  \n",
       "AB_CD45RA                    5.536415          0.999225          1.000000  \n",
       "AB_TCR                       4.048441          1.000000          0.999647  \n",
       "...                               ...               ...               ...  \n",
       "AB_CD71                      1.406471          0.817829          0.810887  \n",
       "AB_HLA-A2                    0.056526          0.051163          0.053022  \n",
       "AB_mouse/ratCD278(ICOS)      2.085696          0.959690          0.952633  \n",
       "AB_CD194(CCR4)               1.898372          0.927907          0.927890  \n",
       "AB_CD26                      1.513948          0.850388          0.856486  \n",
       "\n",
       "[192 rows x 7 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_PS_protein = DE_table_PS[DE_table_PS.index.isin(protein_list)]\n",
    "DE_table_PS_protein.to_csv('/home/jovyan/COVID/NB5_DE Analysis/DE proteins/NK CD56(dim)_DE_table_PS_vs_rest_protein.csv')\n",
    "DE_table_PS_protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "technical-portrait",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_CD62P(P-Selectin)</th>\n",
       "      <td>-1.303985</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>2.086339</td>\n",
       "      <td>3.390324</td>\n",
       "      <td>0.934837</td>\n",
       "      <td>0.994549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD235ab</th>\n",
       "      <td>-1.100213</td>\n",
       "      <td>3.843564e-252</td>\n",
       "      <td>1.017831e-248</td>\n",
       "      <td>0.895412</td>\n",
       "      <td>1.995625</td>\n",
       "      <td>0.623058</td>\n",
       "      <td>0.874823</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD49b</th>\n",
       "      <td>-0.986591</td>\n",
       "      <td>1.749071e-239</td>\n",
       "      <td>3.602503e-236</td>\n",
       "      <td>1.237850</td>\n",
       "      <td>2.224441</td>\n",
       "      <td>0.674687</td>\n",
       "      <td>0.915607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD41</th>\n",
       "      <td>-1.380405</td>\n",
       "      <td>6.983420e-234</td>\n",
       "      <td>1.176833e-230</td>\n",
       "      <td>3.115313</td>\n",
       "      <td>4.495718</td>\n",
       "      <td>0.982957</td>\n",
       "      <td>0.986877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD15(SSEA-1)</th>\n",
       "      <td>-0.850248</td>\n",
       "      <td>1.339410e-224</td>\n",
       "      <td>2.069054e-221</td>\n",
       "      <td>2.490928</td>\n",
       "      <td>3.341176</td>\n",
       "      <td>0.956391</td>\n",
       "      <td>0.994549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD44</th>\n",
       "      <td>0.009306</td>\n",
       "      <td>6.423507e-01</td>\n",
       "      <td>7.344717e-01</td>\n",
       "      <td>0.651334</td>\n",
       "      <td>0.642028</td>\n",
       "      <td>0.483208</td>\n",
       "      <td>0.458308</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_HLA-A2</th>\n",
       "      <td>0.002844</td>\n",
       "      <td>6.641269e-01</td>\n",
       "      <td>7.519038e-01</td>\n",
       "      <td>0.058750</td>\n",
       "      <td>0.055906</td>\n",
       "      <td>0.053634</td>\n",
       "      <td>0.052292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD62L</th>\n",
       "      <td>0.009352</td>\n",
       "      <td>7.068128e-01</td>\n",
       "      <td>7.862995e-01</td>\n",
       "      <td>3.327438</td>\n",
       "      <td>3.318086</td>\n",
       "      <td>0.998496</td>\n",
       "      <td>0.992732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD45</th>\n",
       "      <td>0.001544</td>\n",
       "      <td>9.383365e-01</td>\n",
       "      <td>9.598247e-01</td>\n",
       "      <td>3.085182</td>\n",
       "      <td>3.083638</td>\n",
       "      <td>0.992481</td>\n",
       "      <td>0.989905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD314(NKG2D)</th>\n",
       "      <td>0.000690</td>\n",
       "      <td>9.721001e-01</td>\n",
       "      <td>9.817390e-01</td>\n",
       "      <td>1.977032</td>\n",
       "      <td>1.976342</td>\n",
       "      <td>0.947870</td>\n",
       "      <td>0.944276</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         logFC        P.Value      adj.P.Val  AveExpr_case  \\\n",
       "Gene                                                                         \n",
       "AB_CD62P(P-Selectin) -1.303985   0.000000e+00   0.000000e+00      2.086339   \n",
       "AB_CD235ab           -1.100213  3.843564e-252  1.017831e-248      0.895412   \n",
       "AB_CD49b             -0.986591  1.749071e-239  3.602503e-236      1.237850   \n",
       "AB_CD41              -1.380405  6.983420e-234  1.176833e-230      3.115313   \n",
       "AB_CD15(SSEA-1)      -0.850248  1.339410e-224  2.069054e-221      2.490928   \n",
       "...                        ...            ...            ...           ...   \n",
       "AB_CD44               0.009306   6.423507e-01   7.344717e-01      0.651334   \n",
       "AB_HLA-A2             0.002844   6.641269e-01   7.519038e-01      0.058750   \n",
       "AB_CD62L              0.009352   7.068128e-01   7.862995e-01      3.327438   \n",
       "AB_CD45               0.001544   9.383365e-01   9.598247e-01      3.085182   \n",
       "AB_CD314(NKG2D)       0.000690   9.721001e-01   9.817390e-01      1.977032   \n",
       "\n",
       "                      AveExpr_ctrl  percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                                    \n",
       "AB_CD62P(P-Selectin)      3.390324          0.934837          0.994549  \n",
       "AB_CD235ab                1.995625          0.623058          0.874823  \n",
       "AB_CD49b                  2.224441          0.674687          0.915607  \n",
       "AB_CD41                   4.495718          0.982957          0.986877  \n",
       "AB_CD15(SSEA-1)           3.341176          0.956391          0.994549  \n",
       "...                            ...               ...               ...  \n",
       "AB_CD44                   0.642028          0.483208          0.458308  \n",
       "AB_HLA-A2                 0.055906          0.053634          0.052292  \n",
       "AB_CD62L                  3.318086          0.998496          0.992732  \n",
       "AB_CD45                   3.083638          0.992481          0.989905  \n",
       "AB_CD314(NKG2D)           1.976342          0.947870          0.944276  \n",
       "\n",
       "[192 rows x 7 columns]"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_MS_protein = DE_table_MS[DE_table_MS.index.isin(protein_list)]\n",
    "DE_table_MS_protein.to_csv('/home/jovyan/COVID/NB5_DE Analysis/DE proteins/NK CD56(dim)_DE_table_MS_vs_rest_protein.csv')\n",
    "DE_table_MS_protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "roman-murder",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>logFC</th>\n",
       "      <th>P.Value</th>\n",
       "      <th>adj.P.Val</th>\n",
       "      <th>AveExpr_case</th>\n",
       "      <th>AveExpr_ctrl</th>\n",
       "      <th>percentExpr_case</th>\n",
       "      <th>percentExpr_ctrl</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AB_IgM</th>\n",
       "      <td>-1.479977</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.261228</td>\n",
       "      <td>3.741205</td>\n",
       "      <td>0.979620</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgA</th>\n",
       "      <td>-1.323790</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.538875</td>\n",
       "      <td>3.862665</td>\n",
       "      <td>0.990898</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD235ab</th>\n",
       "      <td>-1.850082</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.175392</td>\n",
       "      <td>3.025474</td>\n",
       "      <td>0.732291</td>\n",
       "      <td>0.989968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_IgGFc</th>\n",
       "      <td>-1.063513</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.400844</td>\n",
       "      <td>4.464357</td>\n",
       "      <td>0.999011</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_Iglightchain-1</th>\n",
       "      <td>-1.074803</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.011597</td>\n",
       "      <td>4.086401</td>\n",
       "      <td>0.998219</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD26</th>\n",
       "      <td>0.011153</td>\n",
       "      <td>0.621642</td>\n",
       "      <td>0.735848</td>\n",
       "      <td>1.517155</td>\n",
       "      <td>1.506002</td>\n",
       "      <td>0.852592</td>\n",
       "      <td>0.862724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD86</th>\n",
       "      <td>-0.008325</td>\n",
       "      <td>0.625702</td>\n",
       "      <td>0.739143</td>\n",
       "      <td>0.404401</td>\n",
       "      <td>0.412725</td>\n",
       "      <td>0.323704</td>\n",
       "      <td>0.327350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD64</th>\n",
       "      <td>0.003193</td>\n",
       "      <td>0.873727</td>\n",
       "      <td>0.920243</td>\n",
       "      <td>0.711348</td>\n",
       "      <td>0.708155</td>\n",
       "      <td>0.524733</td>\n",
       "      <td>0.518479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_TCRa/B</th>\n",
       "      <td>0.001899</td>\n",
       "      <td>0.923633</td>\n",
       "      <td>0.953458</td>\n",
       "      <td>3.042553</td>\n",
       "      <td>3.040654</td>\n",
       "      <td>0.994064</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AB_CD25</th>\n",
       "      <td>-0.001274</td>\n",
       "      <td>0.952268</td>\n",
       "      <td>0.971021</td>\n",
       "      <td>1.061219</td>\n",
       "      <td>1.062493</td>\n",
       "      <td>0.705580</td>\n",
       "      <td>0.717529</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      logFC   P.Value  adj.P.Val  AveExpr_case  AveExpr_ctrl  \\\n",
       "Gene                                                                           \n",
       "AB_IgM            -1.479977  0.000000   0.000000      2.261228      3.741205   \n",
       "AB_IgA            -1.323790  0.000000   0.000000      2.538875      3.862665   \n",
       "AB_CD235ab        -1.850082  0.000000   0.000000      1.175392      3.025474   \n",
       "AB_IgGFc          -1.063513  0.000000   0.000000      3.400844      4.464357   \n",
       "AB_Iglightchain-1 -1.074803  0.000000   0.000000      3.011597      4.086401   \n",
       "...                     ...       ...        ...           ...           ...   \n",
       "AB_CD26            0.011153  0.621642   0.735848      1.517155      1.506002   \n",
       "AB_CD86           -0.008325  0.625702   0.739143      0.404401      0.412725   \n",
       "AB_CD64            0.003193  0.873727   0.920243      0.711348      0.708155   \n",
       "AB_TCRa/B          0.001899  0.923633   0.953458      3.042553      3.040654   \n",
       "AB_CD25           -0.001274  0.952268   0.971021      1.061219      1.062493   \n",
       "\n",
       "                   percentExpr_case  percentExpr_ctrl  \n",
       "Gene                                                   \n",
       "AB_IgM                     0.979620          1.000000  \n",
       "AB_IgA                     0.990898          1.000000  \n",
       "AB_CD235ab                 0.732291          0.989968  \n",
       "AB_IgGFc                   0.999011          1.000000  \n",
       "AB_Iglightchain-1          0.998219          1.000000  \n",
       "...                             ...               ...  \n",
       "AB_CD26                    0.852592          0.862724  \n",
       "AB_CD86                    0.323704          0.327350  \n",
       "AB_CD64                    0.524733          0.518479  \n",
       "AB_TCRa/B                  0.994064          1.000000  \n",
       "AB_CD25                    0.705580          0.717529  \n",
       "\n",
       "[192 rows x 7 columns]"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DE_table_Control_protein = DE_table_Control[DE_table_Control.index.isin(protein_list)]\n",
    "DE_table_Control_protein.to_csv('/home/jovyan/COVID/NB5_DE Analysis/DE proteins/NK CD56(dim)_DE_table_Control_vs_rest_protein.csv')\n",
    "DE_table_Control_protein"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "marked-stuff",
   "metadata": {},
   "source": [
    "End of Notebook"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
